
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../quality_control/">
      
      
        <link rel="next" href="../mapping/">
      
      <link rel="icon" href="../../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.18">
    
    
      
        <title>Sequence trimming - Introduction to RNA-Seq - From quality control to pathway analysis</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.26e3688c.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#material" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-header__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to RNA-Seq - From quality control to pathway analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Sequence trimming
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/RNAseq-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-nav__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to RNA-Seq - From quality control to pathway analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/RNAseq-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../precourse/" class="md-nav__link">
        Precourse preparations
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../course_schedule/" class="md-nav__link">
        Course schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../slides_notes/" class="md-nav__link">
        Slides notes (draft)
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../cheatsheets/" class="md-nav__link">
        Cheatsheets
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
      
      
      
        <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Day1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../design/" class="md-nav__link">
        RNAseq - technologies and design
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../server_login/" class="md-nav__link">
        Server login + unix fresh up
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../quality_control/" class="md-nav__link">
        Quality control
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Sequence trimming
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Sequence trimming
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#to-trim-or-not-to-trim" class="md-nav__link">
    to trim or not to trim ?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trimming-with-trimmomatic" class="md-nav__link">
    trimming with Trimmomatic
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
      
      
      
        <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Day2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../mapping/" class="md-nav__link">
        Reads mapping
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../DE/" class="md-nav__link">
        Differential Expression Inference
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../enrichment/" class="md-nav__link">
        Enrichment analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#to-trim-or-not-to-trim" class="md-nav__link">
    to trim or not to trim ?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trimming-with-trimmomatic" class="md-nav__link">
    trimming with Trimmomatic
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


  <h1>Sequence trimming</h1>

<p>Following a QC analysis on sequencing results, one may detect stretches of low quality bases along reads, or a contamination by adapter sequence.
Depending on your research question and the software you use for mapping, you may have to remove these bad quality / spurious sequences out of your data.</p>
<p><strong>During this block, you will learn to :</strong></p>
<ul>
<li>trim your data with trimmomatic</li>
</ul>
<h2 id="material">Material</h2>
<p><a :=":" class="md-button" href="../../assets/pdf/RNA-Seq_03_trimming.pdf" target="_blank"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0h160v128c0 17.7 14.3 32 32 32h128v144H176c-35.3 0-64 28.7-64 64v144H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0l128 128zM176 352h32c30.9 0 56 25.1 56 56s-25.1 56-56 56h-16v32c0 8.8-7.2 16-16 16s-16-7.2-16-16V368c0-8.8 7.2-16 16-16zm32 80c13.3 0 24-10.7 24-24s-10.7-24-24-24h-16v48h16zm96-80h32c26.5 0 48 21.5 48 48v64c0 26.5-21.5 48-48 48h-32c-8.8 0-16-7.2-16-16V368c0-8.8 7.2-16 16-16zm32 128c8.8 0 16-7.2 16-16v-64c0-8.8-7.2-16-16-16h-16v96h16zm80-112c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16s-7.2 16-16 16h-32v32h32c8.8 0 16 7.2 16 16s-7.2 16-16 16h-32v48c0 8.8-7.2 16-16 16s-16-7.2-16-16V368z"/></svg></span> Download the presentation</a></p>
<p><a :=":" class="md-button" href="http://www.usadellab.org/cms/?page=trimmomatic" target="_blank">Trimmomatic website</a></p>
<h2 id="to-trim-or-not-to-trim">to trim or not to trim ?</h2>
<p>There are several ways to deal with poor quality bases or adapter contamination in reads, and several terms are used in the field, sometimes very loosely. We can talk about:</p>
<ul>
<li><strong>Trimming</strong>: to remove a part of, or the entirety of, a read (for quality reasons).<ul>
<li><strong>Hard trimming</strong>: trim with a high threshold (eg. remove everything with QUAL&lt;30).</li>
<li><strong>Soft trimming</strong>: trim with a low threshold (eg. remove everything with QUAL&lt;10).</li>
</ul>
</li>
<li><strong>Clipping</strong>: to remove the end part of a read (typically because of adapter content).<ul>
<li><strong>Hard clipping</strong>: actually removing the end of the read from the file (ie. with trimmomatic).</li>
<li><strong>Soft clipping</strong>: ignoring the end of the read at mapping time (ie. what STAR does).</li>
</ul>
</li>
</ul>
<p>If the data will be used to perform <strong>transcriptome assembly, or variant analysis, then it MUST be trimmed</strong>.</p>
<p>In contrast, for applications based on <strong>counting reads</strong>, such as <strong>Differential Expression analysis</strong>, most aligners, such as <a href="https://github.com/alexdobin/STAR">STAR</a>, <a href="http://daehwankimlab.github.io/hisat2/">HISAT2</a>, <a href="https://salmon.readthedocs.io/en/latest/salmon.html">salmon</a>, and <a href="https://pachterlab.github.io/kallisto/manual">kallisto</a>, can handle bad quality sequences and adapter content by soft-clipping, and consequently they <em>usually</em> do not need trimming.
In fact, <strong>trimming can be detrimental</strong> to the number of successfully quantified reads [<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-0956-2">William et al. 2016</a>].</p>
<p>Nevertheless, it is usually recommended to perform some amount of soft clipping (<em>eg.</em> <a href="https://www.biostars.org/p/389324/">kallisto</a>, <a href="https://github.com/COMBINE-lab/salmon/issues/398">salmon</a> ).</p>
<p>If possible, we recommend to perform the mapping for both the raw data and the trimmed one, in order to compare the results for both, and choose the best.</p>
<p><strong>Question:</strong> what could be a good metric to choose the best between the trimmed and untrimmed?</p>
<details class="success">
<summary>Answer</summary>
<p>The number of uniquely mapped reads is generally what would matter in differential expression analysis. Of course, this means that you can only choose after you have mapped both the trimmed and the untrimmed reads.</p>
</details>
<h2 id="trimming-with-trimmomatic">trimming with Trimmomatic</h2>
<p>The <a href="http://www.usadellab.org/cms/?page=trimmomatic">trimmomatic website</a> gives very good examples of their software usage for both paired-end (<code>PE</code>) and single-end (<code>SE</code>) reads. We recommend you read their quick-start section attentively.</p>
<p><strong>Task 1:</strong> </p>
<ul>
<li>
<p>Conduct a soft trimming on the mouseMT data</p>
<ul>
<li>name the output folder : <code>030_d_trim/</code>.</li>
<li>when you do <code>ml trimmomatic</code>, you will receive a little message which tells you how to launch the software.</li>
<li>Adapter sequences can be found in <code>/shared/data/DATA/adapters/TruSeq3-PE.fa</code> (<a href="https://github.com/usadellab/Trimmomatic/tree/main/adapters">adapter sequence source</a>).</li>
<li>unlike fastqc, you will have to launch trimmomatic for each sample separately</li>
<li>to facilitate QC afterward, add the following at the end of your trimmomatic command (substituting <code>&lt;sample name&gt;</code>):
            <code>2&gt; 030_trim/trim_out.&lt;sample name&gt;.log</code>
            This will send part of the output of trimmomatic to a file in the same folder as the trimmed reads, which multiQC will be able to use afterward.</li>
</ul>
</li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>trimmomatic is a Java-based program, and thus must be run by passing its .jar file to the Java interpreter:</p>
<div class="highlight"><pre><span></span><code>ml  trimmomatic
java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.39.jar
</code></pre></div>
</div>
<details class="success">
<summary>trimmomatics sbatch script</summary>
<p>We chose the following option:</p>
<ul>
<li><strong>SLIDINGWINDOW:3:25</strong> Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold.</li>
<li><strong>3</strong>  : windowSize: specifies the number of bases to average across</li>
<li><strong>25</strong> : requiredQuality: specifies the average quality required.</li>
<li><strong>ILLUMINACLIP:/shared/home/SHARED/DATA/adapters/TruSeq3-PE.fa:2:30:10</strong> Cut adapter and other Illumina-specific sequences from the read.</li>
<li>Cut adapter and other illumina-specific sequences from the read.</li>
<li><strong>2</strong>  : seedMismatches: specifies the maximum mismatch count which will still allow a full match to be performed</li>
<li><strong>30</strong> : palindromeClipThreshold: specifies how accurate the match between the two &lsquo;adapter ligated&rsquo; reads must be for PE palindrome read alignment.</li>
<li><strong>10</strong> : simpleClipThreshold: specifies how accurate the match between any adapter etc. sequence must be against a read.</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=trim_mouseMT</span>
<span class="c1">#SBATCH --time=01:00:00</span>
<span class="c1">#SBATCH --cpus-per-task=1</span>
<span class="c1">#SBATCH --mem=1G</span>
<span class="c1">#SBATCH -o 030_l_trim_mouseMT.o</span>

ml<span class="w">  </span>trimmomatic

<span class="c1">## creating output folder, in case it does not exists</span>
mkdir<span class="w"> </span>-p<span class="w"> </span>030_d_trim

<span class="c1">## we store the input folder in a variable, </span>
<span class="c1">## to access its value, we will write $INPUT_FOLDER</span>
<span class="nv">INPUT_FOLDER</span><span class="o">=</span>/shared/data/DATA/mouseMT

<span class="c1">## by ending a line with \ we can continue the same command on the line below</span>

java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$EBROOTTRIMMOMATIC</span>/trimmomatic-0.39.jar<span class="w"> </span>SE<span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span><span class="nv">$INPUT_FOLDER</span>/sample_a1.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>030_d_trim/sample_a1.trimmed.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>SLIDINGWINDOW:3:25<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>030_d_trim/030_l_trim_out.sample_a1.log

java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$EBROOTTRIMMOMATIC</span>/trimmomatic-0.39.jar<span class="w"> </span>SE<span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span><span class="nv">$INPUT_FOLDER</span>/sample_a2.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>030_d_trim/sample_a2.trimmed.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>SLIDINGWINDOW:3:25<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>030_d_trim/030_l_trim_out.sample_a2.log

java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$EBROOTTRIMMOMATIC</span>/trimmomatic-0.39.jar<span class="w"> </span>SE<span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span><span class="nv">$INPUT_FOLDER</span>/sample_a3.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>030_d_trim/sample_a3.trimmed.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>SLIDINGWINDOW:3:25<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>030_d_trim/030_l_trim_out.sample_a3.log

java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$EBROOTTRIMMOMATIC</span>/trimmomatic-0.39.jar<span class="w"> </span>SE<span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span><span class="nv">$INPUT_FOLDER</span>/sample_a4.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>030_d_trim/sample_a4.trimmed.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>SLIDINGWINDOW:3:25<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>030_d_trim/030_l_trim_out.sample_a4.log


java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$EBROOTTRIMMOMATIC</span>/trimmomatic-0.39.jar<span class="w"> </span>SE<span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span><span class="nv">$INPUT_FOLDER</span>/sample_b1.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>030_d_trim/sample_b1.trimmed.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>SLIDINGWINDOW:3:25<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>030_d_trim/030_l_trim_out.sample_b1.log

java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$EBROOTTRIMMOMATIC</span>/trimmomatic-0.39.jar<span class="w"> </span>SE<span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span><span class="nv">$INPUT_FOLDER</span>/sample_b2.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>030_d_trim/sample_b2.trimmed.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>SLIDINGWINDOW:3:25<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>030_d_trim/030_l_trim_out.sample_b2.log

java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$EBROOTTRIMMOMATIC</span>/trimmomatic-0.39.jar<span class="w"> </span>SE<span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span><span class="nv">$INPUT_FOLDER</span>/sample_b3.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>030_d_trim/sample_b3.trimmed.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>SLIDINGWINDOW:3:25<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>030_d_trim/030_l_trim_out.sample_b3.log

java<span class="w"> </span>-jar<span class="w"> </span><span class="nv">$EBROOTTRIMMOMATIC</span>/trimmomatic-0.39.jar<span class="w"> </span>SE<span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span><span class="nv">$INPUT_FOLDER</span>/sample_b4.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>030_d_trim/sample_b4.trimmed.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span><span class="se">\</span>
<span class="w">                                         </span>SLIDINGWINDOW:3:25<span class="w"> </span><span class="m">2</span>&gt;<span class="w"> </span>030_d_trim/030_l_trim_out.sample_b4.log
</code></pre></div>
<p>On the cluster, this script is also in <code>/shared/data/Solutions/mouseMT/030_s_trim.sh</code></p>
</details>
<details class="success">
<summary>alternative sbatch using an array job</summary>
<p>First create a file named <code>sampleNames.txt</code>, containing the sample names:</p>
<p><div class="highlight"><pre><span></span><code>sample_a1
sample_a2
sample_a3
sample_a4
sample_b1
sample_b2
sample_b3
sample_b4
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/mouseMT/sampleNames.txt</code></p>
<p><div class="highlight"><pre><span></span><code>#!/usr/bin/bash
#SBATCH --job-name=trim_mouseMT_array
#SBATCH --time=01:00:00
#SBATCH --cpus-per-task=1
#SBATCH --mem=1G
#SBATCH -o 030_l_trim_mouseMT.%a.o
#SBATCH --array=1-8%8

ml  trimmomatic

## creating output folder, in case it does not exists
mkdir -p 030_d_trim

INPUT_FOLDER=/shared/data/DATA/mouseMT

## each job grab a specific line from sampleNames.txt
SAMPLE=`sed -n ${SLURM_ARRAY_TASK_ID}p sampleNames.txt`


java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.39.jar SE -phred33 \
                                         $INPUT_FOLDER/${SAMPLE}.fastq \
                                         030_d_trim/${SAMPLE}.trimmed.fastq \
                                         ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10 \
                                         SLIDINGWINDOW:3:25 2&gt; 030_d_trim/030_l_trim_out.${SAMPLE}.log
</code></pre></div>
On the cluster, this script is also in  <code>/shared/data/Solutions/mouseMT/030bis_s_trim_array.sh</code></p>
</details>
<p><strong>Task 2:</strong> </p>
<ul>
<li>Use the the following script to run a QC analysis on your trimmmed reads and compare with the raw ones.</li>
</ul>
<p><div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=trim-multiqc</span>
<span class="c1">#SBATCH --time=00:30:00</span>
<span class="c1">#SBATCH --cpus-per-task=1</span>
<span class="c1">#SBATCH --mem=1G</span>
<span class="c1">#SBATCH -o 032_l_multiqc_trimmed.o</span>

ml<span class="w"> </span>fastqc
<span class="c1">## fastQC on trimmed fastq files</span>
fastqc<span class="w"> </span>030_d_trim/*.fastq<span class="w"> </span>-o<span class="w"> </span>030_d_trim


<span class="c1">## multiqc on the fastQC reports AND the trimmomatic logs</span>
multiqc<span class="w"> </span>-n<span class="w"> </span>032_r_multiqc_mouseMT_trimmed.html<span class="w"> </span>-f<span class="w"> </span>--title<span class="w"> </span>trimmed_fastq<span class="w"> </span>030_d_trim/
</code></pre></div>
On the cluster, you can find this script in : <code>/shared/data/Solutions/mouseMT/032_s_multiqc_trimmed.sh</code></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The script above presumes that you have successfully trimmed the reads. </p>
<p>If not, you can grab them on the cluster in <code>/shared/data/Solutions/mouseMT/030_d_trim/</code></p>
</div>
<details class="success">
<summary>trimmed reads multiqc report</summary>
<p><a :=":" class="md-button" href="../../assets/html/032_multiqc_mouseMT_trimmed.html" target="_blank"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0h160v128c0 17.7 14.3 32 32 32h128v288c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0l128 128z"/></svg></span> Download the multiqc report</a></p>
<p><strong>Note the second section, Trimmomatic, which lets you know the number/percentage of reads dropped</strong></p>
</details>
<!-- I'd add back the "important notes" -->

<!--
---

Important notes :

 * when you do `ml trimmomatic`, you will receive a little message which tells you how to launch the software.
 * Adapter sequences can be found in `/shared/data/DATA/adapters/`.
 * Trimmomatic RAM requirements : ~0.5G / CPU.
 * Trimmomatic time requirements : ~ 10 min/ read file .
-->

<!-- 
??? done "Trimmomatic script"

    The Liu2015 dataset has paired-end reads and we have to take that into account during trimming.
    For a soft-trimming, we chose the following options :

     * **SLIDINGWINDOW:4:20** Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold.
        - **4**  : windowSize: specifies the number of bases to average across
        - **20** : requiredQuality: specifies the average quality required.
     * **ILLUMINACLIP:/shared/home/SHARED/DATA/adapters/TruSeq3-PE.fa:2:30:10** Cut adapter and other Illumina-specific sequences from the read.
         - Cut adapter and other illumina-specific sequences from the read.
         - **2**  : seedMismatches: specifies the maximum mismatch count which will still allow a full match to be performed
         - **30** : palindromeClipThreshold: specifies how accurate the match between the two 'adapter ligated' reads must be for PE palindrome read alignment.
         - **10** : simpleClipThreshold: specifies how accurate the match between any adapter etc. sequence must be against a read.

    Here is a script for a single sample : 

    <div class="highlight"><pre><span></span><code><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --job-name=trim</span>
<span class="c1">#SBATCH --time=01:00:00</span>
<span class="c1">#SBATCH --cpus-per-task=4</span>
<span class="c1">#SBATCH --mem-per-cpu=4G</span>
<span class="c1">#SBATCH -o trim.o</span>
<span class="c1">#SBATCH -e trim.e</span>

ml<span class="w">  </span>trimmomatic

<span class="nv">dataDIR</span><span class="o">=</span>/shared/data/DATA/Liu2015

<span class="nv">trimmomatic</span><span class="o">=</span><span class="s2">&quot;java -jar </span><span class="nv">$EBROOTTRIMMOMATIC</span><span class="s2">/trimmomatic-0.39.jar&quot;</span>

<span class="nv">outDIR</span><span class="o">=</span>Liu2015_trimmed_reads

mkdir<span class="w"> </span>-p<span class="w"> </span><span class="nv">$outDIR</span>

<span class="nv">$trimmomatic</span><span class="w"> </span>PE<span class="w"> </span>-threads<span class="w"> </span><span class="m">4</span><span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<span class="w">             </span><span class="nv">$dataDIR</span>/SRR1272187_1.fastq.gz<span class="w"> </span><span class="se">\</span>
<span class="w">             </span><span class="nv">$dataDIR</span>/SRR1272187_2.fastq.gz<span class="w"> </span><span class="se">\</span>
<span class="w">             </span><span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_paired_1.fastq<span class="w"> </span><span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_unpaired_1.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">             </span><span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_paired_2.fastq<span class="w"> </span><span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_unpaired_2.fastq<span class="w"> </span><span class="se">\</span>
<span class="w">             </span>SLIDINGWINDOW:4:20<span class="w"> </span>ILLUMINACLIP:/shared/data/DATA/adapters/TruSeq3-PE.fa:2:30:10<span class="w"> </span>

<span class="c1">## compressing the resulting fastq files to save some space.</span>
gzip<span class="w"> </span><span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_paired_1.fastq
gzip<span class="w"> </span><span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_unpaired_1.fastq
gzip<span class="w"> </span><span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_paired_2.fastq
gzip<span class="w"> </span><span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_unpaired_2.fastq
</code></pre></div>

-->





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.74e28a9f.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.220ee61c.min.js"></script>
      
    
  </body>
</html>