
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      <link rel="icon" href="../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.2.1, mkdocs-material-7.1.8">
    
    
      
        <title>Sequence trimming - Introduction to RNA-Seq - From quality control to pathway analysis</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.ca7ac06f.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.f1a3b89f.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    
      


    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#material" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../index.html" title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-header__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to RNA-Seq - From quality control to pathway analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Sequence trimming
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/sib-swiss/RNAseq-introduction-training/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../index.html" title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-nav__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to RNA-Seq - From quality control to pathway analysis
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/sib-swiss/RNAseq-introduction-training/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../index.html" class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../precourse.html" class="md-nav__link">
        Precourse preparations
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../course_schedule.html" class="md-nav__link">
        Course schedule
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" checked>
      
      <label class="md-nav__link" for="__nav_4">
        Day1
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Day1" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Day1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="design.html" class="md-nav__link">
        RNAseq - technologies and design
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="server_login.html" class="md-nav__link">
        Server login + unix fresh up
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="quality_control.html" class="md-nav__link">
        Quality control
      </a>
    </li>
  

          
            
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Sequence trimming
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="trimming.html" class="md-nav__link md-nav__link--active">
        Sequence trimming
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#to-trim-or-not-to-trim" class="md-nav__link">
    to trim or not to trim ?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trimming-with-trimmomatic" class="md-nav__link">
    trimming with Trimmomatic
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="mapping.html" class="md-nav__link">
        Reads mapping
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      <label class="md-nav__link" for="__nav_5">
        Day2
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Day2" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Day2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../day2/counting.html" class="md-nav__link">
        Read counting
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../day2/DE.html" class="md-nav__link">
        Differential Expression Inference
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#to-trim-or-not-to-trim" class="md-nav__link">
    to trim or not to trim ?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trimming-with-trimmomatic" class="md-nav__link">
    trimming with Trimmomatic
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/sib-swiss/RNAseq-introduction-training/edit/master/docs/day1/trimming.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  <h1>Sequence trimming</h1>
                
                <p>Following a QC analysis on sequencing results, one could detect stretches of low quality bases along reads, or a contamination by adapter sequence.
Depending on your application and the sofware you use for mapping, you may have to remove these bad quality / spurious sequences out of your data.</p>
<p>During this block, you will learn to :</p>
<ul>
<li>trim your data with trimmomatic</li>
</ul>
<h2 id="material">Material</h2>
<p><a class="md-button" href="../assets/pdf/RNA-Seq_03_trimming.pdf"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><path d="M181.9 256.1c-5-16-4.9-46.9-2-46.9 8.4 0 7.6 36.9 2 46.9zm-1.7 47.2c-7.7 20.2-17.3 43.3-28.4 62.7 18.3-7 39-17.2 62.9-21.9-12.7-9.6-24.9-23.4-34.5-40.8zM86.1 428.1c0 .8 13.2-5.4 34.9-40.2-6.7 6.3-29.1 24.5-34.9 40.2zM248 160h136v328c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V24C0 10.7 10.7 0 24 0h200v136c0 13.2 10.8 24 24 24zm-8 171.8c-20-12.2-33.3-29-42.7-53.8 4.5-18.5 11.6-46.6 6.2-64.2-4.7-29.4-42.4-26.5-47.8-6.8-5 18.3-.4 44.1 8.1 77-11.6 27.6-28.7 64.6-40.8 85.8-.1 0-.1.1-.2.1-27.1 13.9-73.6 44.5-54.5 68 5.6 6.9 16 10 21.5 10 17.9 0 35.7-18 61.1-61.8 25.8-8.5 54.1-19.1 79-23.2 21.7 11.8 47.1 19.5 64 19.5 29.2 0 31.2-32 19.7-43.4-13.9-13.6-54.3-9.7-73.6-7.2zM377 105 279 7c-4.5-4.5-10.6-7-17-7h-6v128h128v-6.1c0-6.3-2.5-12.4-7-16.9zm-74.1 255.3c4.1-2.7-2.5-11.9-42.8-9 37.1 15.8 42.8 9 42.8 9z"/></svg></span> Download the presentation</a></p>
<p><a class="md-button" href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic website</a></p>
<h2 id="to-trim-or-not-to-trim">to trim or not to trim ?</h2>
<p>If the data will be used to perform <strong>transcriptome assembly, or variant analysis, then it must be trimmed</strong>.</p>
<p>In contrast, for applications based on <strong>counting reads, such as Differential Expression analysis</strong>, most aligners, such as <a href="https://github.com/alexdobin/STAR">STAR</a>, <a href="http://daehwankimlab.github.io/hisat2/">HISAT2</a>, <a href="https://salmon.readthedocs.io/en/latest/salmon.html">salmon</a>, <a href="https://pachterlab.github.io/kallisto/manual">kallisto</a> can handle bad quality sequences and adapter content by performing &ldquo;soft-clipping&rdquo; on reads, and consequently they <strong>usually do not need trimming.
In fact, hard trimming can be detrimental</strong> to the number of successfully quantified reads [<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-0956-2">William et al. 2016</a>].</p>
<p>Nevertheless, they usually recommend to perform some amount of soft trimming (<em>eg.</em> <a href="https://www.biostars.org/p/389324/">kallisto</a>, <a href="https://github.com/COMBINE-lab/salmon/issues/398">salmon</a> ).
If possible, we recommend to perform the mapping for both the raw data and the trimmed one, in order to compare the results for both and choose the best.</p>
<h2 id="trimming-with-trimmomatic">trimming with Trimmomatic</h2>
<p>The <a href="http://www.usadellab.org/cms/?page=trimmomatic">trimmomatic website</a> gives very good examples of their software usage for both paired-end (<code>PE</code>) and single-end (<code>SE</code>) reads. We recommend you read their quick-start section attentively.</p>
<p><strong>Task :</strong> Conduct a soft trimming on the Liu2015 data.</p>
<p><strong>Extra (if you have the time) :</strong> run a QC analysis on your trimmmed reads and compare with the raw ones.</p>
<p>Important notes :</p>
<ul>
<li>Adapter sequences can be found in <code>/shared/data/DATA/adapters/</code></li>
<li>Trimmomatic RAM requirements : ~0.5G / cpu</li>
<li>Trimmomatic time requirements : ~ 10 min/ read file </li>
</ul>
<details class="done"><summary>Trimmomatic script</summary><p>The Liu2015 dataset has paired-end reads and we have to take that into account during trimming.
For a soft-trimming, we chose the following options :</p>
<ul>
<li><strong>SLIDINGWINDOW:4:20</strong> Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold.<ul>
<li><strong>4</strong>  : windowSize: specifies the number of bases to average across</li>
<li><strong>20</strong> : requiredQuality: specifies the average quality required.</li>
</ul>
</li>
<li><strong>ILLUMINACLIP:/shared/home/SHARED/DATA/adapters/TruSeq3-PE.fa:2:30:10</strong> Cut adapter and other illumina-specific sequences from the read.<ul>
<li>Cut adapter and other illumina-specific sequences from the read.</li>
<li><strong>2</strong>  : seedMismatches: specifies the maximum mismatch count which will still allow a full match to be performed</li>
<li><strong>30</strong> : palindromeClipThreshold: specifies how accurate the match between the two &lsquo;adapter ligated&rsquo; reads must be for PE palindrome read alignment.</li>
<li><strong>10</strong> : simpleClipThreshold: specifies how accurate the match between any adapter etc. sequence must be against a read.</li>
</ul>
</li>
</ul>
<p>Here is a script for a single sample : </p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --job-name=trim</span>
<span class="c1">#SBATCH --time=01:00:00</span>
<span class="c1">#SBATCH --cpus-per-task=4</span>
<span class="c1">#SBATCH --mem-per-cpu=4G</span>
<span class="c1">#SBATCH -o trim.o</span>
<span class="c1">#SBATCH -e trim.e</span>

ml  trimmomatic

<span class="nv">dataDIR</span><span class="o">=</span>/shared/data/DATA/Liu2015

<span class="nv">trimmomatic</span><span class="o">=</span><span class="s2">&quot;java -jar </span><span class="nv">$EBROOTTRIMMOMATIC</span><span class="s2">/trimmomatic-0.39.jar&quot;</span>

<span class="nv">outDIR</span><span class="o">=</span>Liu2015_trimmed_reads

mkdir -p <span class="nv">$outDIR</span>

<span class="nv">$trimmomatic</span> PE -threads <span class="m">4</span> -phred33 <span class="se">\</span>
             <span class="nv">$dataDIR</span>/SRR1272187_1.fastq.gz <span class="se">\</span>
             <span class="nv">$dataDIR</span>/SRR1272187_2.fastq.gz <span class="se">\</span>
             <span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_paired_1.fastq <span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_unpaired_1.fastq <span class="se">\</span>
             <span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_paired_2.fastq <span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_unpaired_2.fastq <span class="se">\</span>
             SLIDINGWINDOW:4:20 ILLUMINACLIP:/adapters/TruSeq3-PE.fa:2:30:10 

<span class="c1">## compressing the resulting fastq files to save some space.</span>
gzip <span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_paired_1.fastq
gzip <span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_unpaired_1.fastq
gzip <span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_paired_2.fastq
gzip <span class="nv">$outDIR</span>/SRR1272187_NFLV_trimmed_unpaired_2.fastq
</code></pre></div>
</details>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="quality_control.html" class="md-footer__link md-footer__link--prev" aria-label="Previous: Quality control" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Quality control
            </div>
          </div>
        </a>
      
      
        
        <a href="mapping.html" class="md-footer__link md-footer__link--next" aria-label="Next: Reads mapping" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Reads mapping
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../assets/javascripts/workers/search.b0710199.min.js", "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.76f349be.min.js"></script>
      
    
  </body>
</html>