
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../course_schedule/">
      
      
        <link rel="next" href="../cheatsheets/">
      
      <link rel="icon" href="../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.18">
    
    
      
        <title>Slides notes (draft) - Introduction to RNA-Seq - From quality control to pathway analysis</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.26e3688c.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#slides-notes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-header__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to RNA-Seq - From quality control to pathway analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Slides notes (draft)
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/RNAseq-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-nav__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to RNA-Seq - From quality control to pathway analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/RNAseq-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../precourse/" class="md-nav__link">
        Precourse preparations
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../course_schedule/" class="md-nav__link">
        Course schedule
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Slides notes (draft)
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Slides notes (draft)
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#01-overview" class="md-nav__link">
    01 Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#02-quality-control" class="md-nav__link">
    02 Quality control
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#03-trimming" class="md-nav__link">
    03 trimming
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#04-mapping" class="md-nav__link">
    04 mapping
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#05-de" class="md-nav__link">
    05 DE
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#06-enrichment" class="md-nav__link">
    06 Enrichment
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../cheatsheets/" class="md-nav__link">
        Cheatsheets
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
      
      
      
        <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Day1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../days/design/" class="md-nav__link">
        RNAseq - technologies and design
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../days/server_login/" class="md-nav__link">
        Server login + unix fresh up
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../days/quality_control/" class="md-nav__link">
        Quality control
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../days/trimming/" class="md-nav__link">
        Sequence trimming
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
      
      
      
        <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Day2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../days/mapping/" class="md-nav__link">
        Reads mapping
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../days/DE/" class="md-nav__link">
        Differential Expression Inference
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../days/enrichment/" class="md-nav__link">
        Enrichment analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#01-overview" class="md-nav__link">
    01 Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#02-quality-control" class="md-nav__link">
    02 Quality control
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#03-trimming" class="md-nav__link">
    03 trimming
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#04-mapping" class="md-nav__link">
    04 mapping
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#05-de" class="md-nav__link">
    05 DE
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#06-enrichment" class="md-nav__link">
    06 Enrichment
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="slides-notes">Slides notes</h1>
<p>This document contains notes about the course slides.</p>
<p>This is a somewhat internal document, so expect a fairly draft-ish and concise style.</p>
<h2 id="01-overview">01 Overview</h2>
<p>slides 3 - 7</p>
<ul>
<li>the central dogma of molecular biology is known to be not so simple</li>
<li>RNA is not only a messenger but may have other roles</li>
<li>most of these elements interacts and regulates one-another </li>
<li>alternative splicing in eukaryots adds a layer of possibilities to all this</li>
<li>main takeaway maybe : measuring RNA is a proxy for protein levels, which is a proxy for protein activity , which is a proxy for the physiological state of the cell</li>
</ul>
<p>slides 8 - 9</p>
<ul>
<li>non-exhaustive list of sequencing possibilities : <a href="https://liorpachter.wordpress.com/seq/">list on Lior Pachter&rsquo;s blog</a></li>
</ul>
<p>slides 10-12</p>
<ul>
<li>RNAseq, the challenges</li>
<li>slide 10 : from human gff, includes ncRNAs</li>
<li>
<p>slide 11 : </p>
<ul>
<li>data source: <a href="https://gtexportal.org/home/datasets">gTex</a> (V8 gene TPMs)</li>
<li>left: 1 sample -&gt; from 1 to $10^5$ TPM </li>
<li>right: 50 random samples -&gt; 10% of genes contribute 90% of the transcripts</li>
<li>NB: mammalian cell 10-30pg RNA/cell , around 360 000 mRNA molecules (<a href="https://www.qiagen.com/us/resources/faq?id=06a192c2-e72d-42e8-9b40-3171e1eb4cb8&amp;lang=en">source</a> )</li>
</ul>
</li>
<li>
<p>slide 12 : important considerations as well</p>
</li>
</ul>
<p>slide 13</p>
<ul>
<li>Illumina : market leader 50-600bp (generally 50-100), 0.1 (nextseq) to 3 (Hiseq) billion reads</li>
<li>Ion torrent : 600bp, 260M reads</li>
<li>Pacbio : 10-30kb N50 , 4M CCS reads </li>
<li>nanopore : theory single molecule, practice: variable (N50 &gt;100kb on ultra-long kit, up to 4.2Mb )</li>
</ul>
<p>slides 14-22 : describe different technologies</p>
<p>Ion Torrent :</p>
<ul>
<li>cell sequentially flooded with A T G C </li>
</ul>
<p>PacBio SMRT</p>
<ul>
<li>DNApol at bottom of Zero-Mode-Waveguide </li>
<li>fluorescent dye on dNTPs</li>
</ul>
<p>Illumina seq :</p>
<ul>
<li>formation of clusters with the same sequence </li>
<li>SBS : labelled nucleotides have reversible terminators, so only 1 base is incorporated at a time.</li>
<li>slide 23: <a href="https://www.france-genomique.org/technological-expertises/whole-genome/sequencage-a-courtes-lectures-par-clusterisation/?lang=en">paired end sequencing</a></li>
<li>slide 24: <a href="https://www.ecseq.com/support/ngs/how-do-strand-specific-sequencing-protocols-work">stranded sequencing</a></li>
<li>slide 25-26: RIN , RNA purification</li>
<li>slide 27-34: sequencing depth and replicates</li>
<li>slide 33-34: this pattern applies to low- mid- and high- expressors (see their supp doc)</li>
<li>slide 35-42: schematic analysis</li>
<li>slide 35 : before the sequencing</li>
<li>slide 36 : basic analysis</li>
<li>slide 37 : basic analysis with trimmed reads</li>
<li>slide 38 : main QC steps</li>
<li>slide 39 : QC steps provide feedback on previous steps</li>
<li>slide 40 : analysis for variant calling / isoform descriptions / &hellip;</li>
<li>slide 41 : when no reference genome: de novo assembly</li>
<li>slide 42 : the analysis which we&rsquo;ll do during this course</li>
</ul>
<h2 id="02-quality-control">02 Quality control</h2>
<p>slide 04 : <a href="https://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/FileFormat_FASTQ-files_swBS.htm">illumina doc</a></p>
<ul>
<li>control bit: 0 when none of the control bits are on, otherwise it is an even number. On HiSeq X and NextSeq systems, control specification is not performed and this number is always 0.</li>
</ul>
<p>slide 16-&hellip; : interpretation of fastQC report</p>
<h2 id="03-trimming">03 trimming</h2>
<ul>
<li>slide 02-05 : enumerating reasons we may want to trim</li>
<li>slide 04 : adapter can be present if, for instance, insert size is shorter than sequenced length. cf. <a href="https://www.ecseq.com/support/ngs/trimming-adapter-sequences-is-it-necessary">this post</a></li>
<li>slide 06-11 : different cases where we may trim</li>
</ul>
<h2 id="04-mapping">04 mapping</h2>
<p>slides 4-8:</p>
<ul>
<li><a href="https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">bowtie2</a></li>
<li><a href="https://github.com/alexdobin/STAR">STAR</a></li>
<li><a href="https://hcc.unl.edu/docs/applications/app_specific/bioinformatics_tools/alignment_tools/tophat_tophat2/">tophat2</a></li>
<li><a href="https://github.com/deweylab/RSEM">RSEM</a></li>
<li><a href="http://cole-trapnell-lab.github.io/cufflinks/">cufflinks</a></li>
<li><a href="https://salmon.readthedocs.io/en/latest/salmon.html">salmon</a></li>
<li><a href="https://pachterlab.github.io/kallisto/">kallisto</a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/tximport.html">tximport</a></li>
<li><a href="https://subread.sourceforge.net/featureCounts.html">featurecount</a></li>
<li><a href="http://ccb.jhu.edu/software/stringtie/index.shtml?t=manual">stringtie</a></li>
<li><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531192-RNAseq-short-variant-discovery-SNPs-Indels-">GATK variant calling pipeline</a></li>
</ul>
<h2 id="05-de">05 DE</h2>
<ul>
<li>slide 3-9 : challenges for RNAseq </li>
<li>slide 4: sequencing depth varies accross libraries<ul>
<li>left : 100 samples from the <a href="https://gtexportal.org/home/datasets">Gtex V8 dataset</a></li>
<li>right : samples from a random binomial with and without a library size factor applied</li>
</ul>
</li>
<li>slide 5: most of the expression is taken by very few genes + a lot of genes have 0 reads  (plots: data from 100 samples from the <a href="https://gtexportal.org/home/datasets">Gtex V8 dataset</a> )</li>
<li>slide 6: small number of samples. 10k simulations of negative binomial draws</li>
<li>slide 7-9 : <a href="xkcd.com/882">xkcd.com/882</a></li>
<li>slide 10-11 : input</li>
<li>slide 12-14 : very good <a href="https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/">blog post on RPKM and TPM</a></li>
<li>slide 15: filtering: <ul>
<li><a href="https://academic.oup.com/bioinformatics/article/29/17/2146/240530">image source</a> max and mean refer to CPM thresholds ; CPM 1: genes with a CPM less than one in more than half the samples are filtered</li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#indfilt">DESeq2 filtering</a></li>
<li>egdeR filtering: section 2.7 of <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">edgeR doc</a></li>
</ul>
</li>
</ul>
<p>slide 16 : <a href="https://pubmed.ncbi.nlm.nih.gov/22988256/">source</a></p>
<div class="codehilite"><pre><span></span><code>* TC: Total count (CPM) - UQ: Upper Quartile - Med: median - Q: quantile
* top left: coef of variation in housekeeping genes in H. sapiens data
* top right: average false-positive rate over 10 independent datasets simulated with varying proportions of differentially expressed genes (from 0% to 30% for each normalization method). 
* bottom:
    * distribution: distribution inter samples look the same
    * Intra-variance: intra group variance 
    * Housekeeping : coef of variation in 30 housekeeping genes, which are presumed to be similarly expressed across conditions
    * clustering: similarity of DE genes with other methods
    * false positive rate : see above
</code></pre></div>

<p>slide 17: normalization</p>
<p>From this <a href="https://www.biostars.org/p/284775/">biostar post</a></p>
<p>EdgeR: Trimmed Mean of M-values (TMM):</p>
<blockquote>
<p>Based on the hypothesis that most genes are not DE. </p>
<p>The TMM factor is computed for each lane, with one lane being considered as a reference sample and the others as test samples. 
For each test sample, TMM is computed as the weighted mean of log ratios between this test and the reference, after exclusion of the most expressed genes and the genes with the largest log ratios. </p>
<p>According to the hypothesis of low DE, this TMM should be close to 1. If it is not, its value provides an estimate of the correction factor that must be applied to the library sizes (and not the raw counts) in order to fulfill the hypothesis. 
[source: https://www.ncbi.nlm.nih.gov/pubmed/22988256]</p>
</blockquote>
<p>DESeq2</p>
<blockquote>
<p>DESeq:  is based on the hypothesis that most genes are not DE. </p>
<div class="codehilite"><pre><span></span><code>   the median of the ratio, for each gene, of its read count over its geometric mean across all lanes.

   The underlying idea is that non-DE genes should have similar read counts across samples, leading to a ratio of 1.

   Assuming most genes are not DE, the median of this ratio for the lane provides an estimate of the correction factor that should be applied to all read counts of this lane to fulfill the hypothesis.

       [source: https://www.ncbi.nlm.nih.gov/pubmed/22988256]
</code></pre></div>

</blockquote>
<p>slide 19: NB model</p>
<div class="codehilite"><pre><span></span><code>* [image source](https://doi.org/10.1186/gb-2010-11-10-r106)
* orange line is the fit w(q)
* purple line show the variance implied by the Poisson distribution 
* dashed orange line is the variance estimate used by edgeR.
</code></pre></div>

<h2 id="06-enrichment">06 Enrichment</h2>
<ul>
<li>slide 08: <a href="http://geneontology.org">geneontology</a></li>
<li>slide 09: <a href="http://reactome.org">reactome</a></li>
<li>slide 10: <a href="http://www.gsea-msigdb.org/gsea/msigdb/index.jsp">GSEA-msigdb</a></li>
<li>
<p>slide 11: <a href="https://www.genome.jp/kegg/">KEGG</a></p>
</li>
<li>
<p>slides 15 to 21 : GSEA </p>
</li>
</ul>
<p>GSEA is used a lot, but it comes in many flavour, with a large number of options
and it is not always easy to understand what is happening.</p>
<blockquote>
<p>&ldquo;The enrichment score (ES) represents the degree to which a set S is over-represented at the top or bottom of the ranked list L. The score is calculated by walking down the list L, increasing a running-sum statistic when we encounter a gene in S and decreasing when it is not encountered. The magnitude of the increment depends on the gene statistics (e.g., correlation of the gene with phenotype). The ES is the maximum deviation from zero encountered in the random walk; it corresponds to a weighted Kolmogorov-Smirnov(KS)-like statistic (Subramanian et al. 2005).&rdquo; (from <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/enrichment-overview.html#gsea-algorithm">clusterProfiler book</a>)</p>
</blockquote>
<p>Then the ES is normalized (NES) in order to compute a p-value.</p>
<ul>
<li>
<p>in the base method (<a href="https://www.pnas.org/doi/full/10.1073/pnas.0506580102">Subramanian 2005</a>) for each gene sets they create a number of permutated dataset for which they compute an ES, and they then compare the ES on the original data to the distribution of permutated ES for that set. 2 flavours of permutation are described : &ldquo;sample permutation&rdquo; and &ldquo;gene permutation&rdquo;. </p>
<ul>
<li>The sample permutation is only possible if the expression data for each sample is given to the method. It is recommended to have at least 7 samples per condition for it to make sense. </li>
<li>The gene permutation is performed from the ranking metric directly. Hence it is sometimes called &ldquo;preranked-GSEA&rdquo; and to my knowledge this is the most often used permutation scheme of the 2.</li>
</ul>
</li>
<li>
<p>in fGSEA (<a href="https://www.biorxiv.org/content/10.1101/060012v3">Korotkevich et al. 2021</a>) several tricks are used to make the p-value computation of &ldquo;preranked-GSEA&rdquo; faster and more accurate for low p-values. Consequently it has become the default GSEA method in some libraries such as clusterProfiler.</p>
</li>
</ul>
<p>Finally, a parameter to consider when performing (preranked-)(f)GSEA is which metric to use to rank genes. 
The most commons are logFC , -log10(pvalues) * logFC , or some signed test statistics (eg. t-test or Wald statistic).</p>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.74e28a9f.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.220ee61c.min.js"></script>
      
    
  </body>
</html>