
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../trimming/">
      
      
        <link rel="next" href="../DE/">
      
      <link rel="icon" href="../../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.18">
    
    
      
        <title>Reads mapping - Introduction to RNA-Seq - From quality control to pathway analysis</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.26e3688c.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#material" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-header__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to RNA-Seq - From quality control to pathway analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Reads mapping
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/RNAseq-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-nav__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to RNA-Seq - From quality control to pathway analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/RNAseq-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../precourse/" class="md-nav__link">
        Precourse preparations
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../course_schedule/" class="md-nav__link">
        Course schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../slides_notes/" class="md-nav__link">
        Slides notes (draft)
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../cheatsheets/" class="md-nav__link">
        Cheatsheets
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
      
      
      
        <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Day1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../design/" class="md-nav__link">
        RNAseq - technologies and design
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../server_login/" class="md-nav__link">
        Server login + unix fresh up
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../quality_control/" class="md-nav__link">
        Quality control
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../trimming/" class="md-nav__link">
        Sequence trimming
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" checked>
      
      
      
        <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Day2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Reads mapping
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Reads mapping
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#building-a-reference-genome-index" class="md-nav__link">
    Building a reference genome index
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mapping-reads-onto-the-reference" class="md-nav__link">
    Mapping reads onto the reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qc-on-the-aligned-reads" class="md-nav__link">
    QC on the aligned reads
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comparison-of-mapping-the-trimmed-reads" class="md-nav__link">
    Comparison of mapping the trimmed reads
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-pseudo-aligning-with-salmon" class="md-nav__link">
    ADDITIONAL : pseudo-aligning with salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-mapping-reads-from-ruhland2016-on-the-reference" class="md-nav__link">
    ADDITIONAL Mapping reads from Ruhland2016 on the reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additionnal-star-2-pass" class="md-nav__link">
    ADDITIONNAL : STAR 2-Pass
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../DE/" class="md-nav__link">
        Differential Expression Inference
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../enrichment/" class="md-nav__link">
        Enrichment analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#building-a-reference-genome-index" class="md-nav__link">
    Building a reference genome index
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mapping-reads-onto-the-reference" class="md-nav__link">
    Mapping reads onto the reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qc-on-the-aligned-reads" class="md-nav__link">
    QC on the aligned reads
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comparison-of-mapping-the-trimmed-reads" class="md-nav__link">
    Comparison of mapping the trimmed reads
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-pseudo-aligning-with-salmon" class="md-nav__link">
    ADDITIONAL : pseudo-aligning with salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-mapping-reads-from-ruhland2016-on-the-reference" class="md-nav__link">
    ADDITIONAL Mapping reads from Ruhland2016 on the reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additionnal-star-2-pass" class="md-nav__link">
    ADDITIONNAL : STAR 2-Pass
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


  <h1>Reads mapping</h1>

<p>Once you are happy with your read sequences in your FASTQ files, you can use a mapper software to align the reads to the genome and thereby find where they originated from.</p>
<p><strong>At the end of this lesson, you will be able to :</strong></p>
<ul>
<li>identify the differences between a local aligner and a pseudo aligner.</li>
<li>perform genome indexing appropriate to your data.</li>
<li>map your RNA-seq data onto a genome.</li>
</ul>
<h2 id="material">Material</h2>
<p><a :=":" class="md-button" href="../../assets/pdf/RNA-Seq_04_Mapping.pdf" target="_blank"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0h160v128c0 17.7 14.3 32 32 32h128v144H176c-35.3 0-64 28.7-64 64v144H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0l128 128zM176 352h32c30.9 0 56 25.1 56 56s-25.1 56-56 56h-16v32c0 8.8-7.2 16-16 16s-16-7.2-16-16V368c0-8.8 7.2-16 16-16zm32 80c13.3 0 24-10.7 24-24s-10.7-24-24-24h-16v48h16zm96-80h32c26.5 0 48 21.5 48 48v64c0 26.5-21.5 48-48 48h-32c-8.8 0-16-7.2-16-16V368c0-8.8 7.2-16 16-16zm32 128c8.8 0 16-7.2 16-16v-64c0-8.8-7.2-16-16-16h-16v96h16zm80-112c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16s-7.2 16-16 16h-32v32h32c8.8 0 16 7.2 16 16s-7.2 16-16 16h-32v48c0 8.8-7.2 16-16 16s-16-7.2-16-16V368z"/></svg></span> Download the presentation</a></p>
<p><a :=":" class="md-button" href="https://github.com/alexdobin/STAR" target="_blank">STAR website</a></p>
<h2 id="building-a-reference-genome-index">Building a reference genome index</h2>
<p>Before any mapping can be achieved, you must first <em>index</em> the genome want to map to. </p>
<p>To do this with STAR, you need two files:</p>
<ul>
<li>a <em>fasta</em> file containing the sequences of the chromosome (or genome contigs)</li>
<li>a <em>gtf</em> file containing annotations (ie. where the genes and exons are)</li>
</ul>
<p>We will be using the Ensembl references, with their accompanying GTF annotations.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>While the data are already on the server here, in practice or if you are following this course without a teacher,
you can grab the reference genome data from the <a href="https://www.ensembl.org/info/data/ftp/index.html">Ensembl ftp website</a>.</p>
<p>In particular, you will want a mouse <a href="http://ftp.ensembl.org/pub/current_fasta/mus_musculus/dna/">DNA fasta file</a> and <a href="http://ftp.ensembl.org/pub/current_gtf/mus_musculus/">gtf file</a>.</p>
</div>
<p>Take note of the genome sequence and annotation versions, you will need this in your paper&rsquo;s methods section!</p>
<p><strong>Task :</strong> Using STAR, build a genome index for the mouse mitochondrial chromosome.</p>
<ul>
<li>.fasta and .gtf files are in : <code>/shared/data/DATA/Mouse_MT_genome/</code>.</li>
<li>create the index in the folder <code>041_d_STAR_mouseMT_reference</code></li>
<li>the module name for this aligner is <code>star</code>.</li>
<li>this job should require less than 4Gb and 10min to run. </li>
</ul>
<div class="admonition info">
<p class="admonition-title">STAR basic parameter for genome index generation</p>
<p>From the <a href="https://raw.githubusercontent.com/alexdobin/STAR/master/doc/STARmanual.pdf">manual</a>. Refer to it for more details</p>
<ul>
<li><code>--runMode genomeGenerate</code> : running STAR in index generation mode</li>
<li><code>--genomeDir &lt;/path/to/genomeDir&gt;</code> : output folder for the index</li>
<li><code>--genomeFastaFiles &lt;/path/to/genome/fasta1&gt;</code> : chromosome sequences fasta file (can be several files)</li>
<li><code>--sjdbGTFfile &lt;/path/to/annotations.gtf&gt;</code> : annotation gtf file</li>
<li><code>--runThreadN &lt;NumberOfThreads&gt;</code> : number of threads to run on </li>
<li><code>--sjdbOverhang &lt;ReadLength-1&gt;</code> : length of the genomic sequence around the annotated junctions to be used in constructing the splice junctions database. Ideally : read length - 1.</li>
</ul>
<p>Additionally, because the genome is so small here (we only use the mitochondrial chromosome after all), you will need the following advanced option:</p>
<ul>
<li><code>--genomeSAindexNbases 5</code> : must be scaled to <code>min(14, log2(GenomeLength)/2 - 1)</code>, so 5 in our case</li>
</ul>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>While your indexing job is running, you can read ahead in STAR&rsquo;s manual to prepare the next step : mapping your reads onto the indexed reference genome.</p>
</div>
<details class="success">
<summary>STAR indexing script</summary>
<div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=star-build</span>
<span class="c1">#SBATCH --time=00:30:00</span>
<span class="c1">#SBATCH --cpus-per-task=2</span>
<span class="c1">#SBATCH --mem=3G</span>
<span class="c1">#SBATCH -o 041_l_star_index.o</span>


<span class="nv">G_FASTA</span><span class="o">=</span>/shared/data/DATA/Mouse_MT_genome/Mus_musculus.GRCm39.dna.chromosome.MT.fa
<span class="nv">G_GTF</span><span class="o">=</span>/shared/data/DATA/Mouse_MT_genome/Mus_musculus.GRCm39.MT.gtf

ml<span class="w"> </span>star

mkdir<span class="w"> </span>-p<span class="w"> </span>041_d_STAR_mouseMT_reference

STAR<span class="w"> </span>--runMode<span class="w"> </span>genomeGenerate<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--genomeDir<span class="w"> </span>041_d_STAR_mouseMT_reference<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--genomeFastaFiles<span class="w"> </span><span class="nv">$G_FASTA</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--sjdbGTFfile<span class="w"> </span><span class="nv">$G_GTF</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--runThreadN<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--genomeSAindexNbases<span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--sjdbOverhang<span class="w"> </span><span class="m">99</span><span class="w"> </span>
</code></pre></div>
<p>It can be found on the cluster at <code>/shared/data/Solutions/mouseMT/041_s_star_index.sh</code></p>
</details>
<p><strong>Extra task :</strong> Determine how you would add an additional feature to your reference, for example for a novel transcript not described by the standard reference.</p>
<details class="success">
<summary>Answer</summary>
<p>You would edit the gtf file to add your additional feature(s), following the <a href="https://www.ensembl.org/info/website/upload/gff.html">proper format</a>.</p>
</details>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>In case you&rsquo;ve got multiple FASTA files for your genome (eg, 1 per chromosome), you may just list them with the <code>genomeFastaFiles</code> option as follow:</p>
<p><code>--genomeFastaFiles /path/to/genome/fasta1.fa /path/to/genome/fasta2.fa /path/to/genome/fasta3.fa ...</code></p>
</div>
<h2 id="mapping-reads-onto-the-reference">Mapping reads onto the reference</h2>
<p><strong>Task :</strong> Using STAR, align the raw FASTQ files of the mouseMT dataset against the mouse mitochondrial reference you just created.</p>
<ul>
<li>if were not able to complete the previous task, you can use the index in <code>/shared/data/Solutions/mouseMT/041_d_STAR_mouseMT_reference</code> .</li>
<li>search the STAR manual for the option to output a BAM file sorted by coordinate.</li>
<li>search the STAR manual for the option to output a geneCounts file.</li>
<li>put the results in folder <code>042_d_STAR_map_raw/</code> .</li>
</ul>
<div class="admonition info">
<p class="admonition-title">STAR basic parameters for mapping</p>
<p>Taken again from the manual:</p>
<ul>
<li><code>--genomeDir &lt;/path/to/genomeDir&gt;</code> : folder where you have put the genome index</li>
<li><code>--readFilesIn &lt;/path/to/read1&gt;</code> : path to a fastq file. If the reads are paired, then also include the path to the second fastq file</li>
<li><code>--runThreadN &lt;NumberOfThreads&gt;</code>: number of threads to run on.</li>
<li><code>--outFileNamePrefix  &lt;prefix&gt;</code> : prefix of the output files, typically something like <code>output_directory/sampleName</code> . </li>
</ul>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Take the time to read the parts of the <a href="https://raw.githubusercontent.com/alexdobin/STAR/master/doc/STARmanual.pdf">STAR manual</a> which concern you: a bit of planning ahead can save you a lot of time-consuming/headache-inducing trial-and-error on your script.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Mapping reads and generating a sorted BAM from one of the mouseMT FASTQ file will take less than a minute and very little RAM, but on a real dataset it should take from 15 minutes to an hour per sample and require at least 30GB of RAM.</p>
</div>
<details class="success">
<summary>STAR mapping script</summary>
<p>We will be using a job array to map each file in different job that will run at the same time.</p>
<p>First create a file named <code>sampleNames.txt</code>, containing the sample names:</p>
<p><div class="highlight"><pre><span></span><code>sample_a1
sample_a2
sample_a3
sample_a4
sample_b1
sample_b2
sample_b3
sample_b4
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/mouseMT/sampleNames.txt</code></p>
<p>Then for our script:</p>
<p><div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=star-aln</span>
<span class="c1">#SBATCH --time=00:10:00</span>
<span class="c1">#SBATCH --cpus-per-task=2</span>
<span class="c1">#SBATCH --mem=1G</span>
<span class="c1">#SBATCH -o 042_l_STAR_map_raw.%a.o</span>
<span class="c1">#SBATCH --array 1-8%8</span>

ml<span class="w"> </span>star

mkdir<span class="w"> </span>-p<span class="w"> </span>042_d_STAR_map_raw

<span class="nv">SAMPLE</span><span class="o">=</span><span class="sb">`</span>sed<span class="w"> </span>-n<span class="w"> </span><span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>p<span class="w"> </span>sampleNames.txt<span class="sb">`</span>

<span class="nv">FASTQ_NAME</span><span class="o">=</span>/shared/data/DATA/mouseMT/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>.fastq

STAR<span class="w"> </span>--runThreadN<span class="w"> </span><span class="m">4</span><span class="w"> </span>--genomeDir<span class="w"> </span>041_d_STAR_mouseMT_reference<span class="w"> </span><span class="se">\</span>
<span class="w">              </span>--outSAMtype<span class="w"> </span>BAM<span class="w"> </span>SortedByCoordinate<span class="w"> </span><span class="se">\</span>
<span class="w">              </span>--outFileNamePrefix<span class="w">  </span>042_d_STAR_map_raw/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>.<span class="w"> </span><span class="se">\</span>
<span class="w">              </span>--quantMode<span class="w"> </span>GeneCounts<span class="w"> </span><span class="se">\</span>
<span class="w">              </span>--readFilesIn<span class="w"> </span><span class="nv">$FASTQ_NAME</span>
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/mouseMT/042_s_STAR_map_raw.sh</code></p>
<p>and its results can be found at <code>/shared/data/Solutions/mouseMT/042_d_STAR_map_raw/</code></p>
<p>The options of STAR are :</p>
<ul>
<li><code>--runThreadN 4</code> : 4 threads to go faster.</li>
<li><code>--genomeDir 041_STAR_reference</code> : path of the genome to map to.</li>
<li><code>--outSAMtype BAM SortedByCoordinate</code> : output a coordinate-sorted BAM file.</li>
<li><code>--outFileNamePrefix 042_STAR_map_raw/${SAMPLE}.</code> : prefix of output files.</li>
<li><code>--quantMode GeneCounts</code> : will create a file with counts of reads per gene.</li>
<li><code>--readFilesIn $FASTQ_NAME</code> : input read file.</li>
</ul>
</details>
<h2 id="qc-on-the-aligned-reads">QC on the aligned reads</h2>
<p>You can call MultiQC on the STAR output folder to gather a report on the individual alignments.</p>
<p><strong>Task :</strong> use <code>multiqc</code> to generate a QC report on the results of your mapping.</p>
<ul>
<li>Evaluate the alignment statistics. Do you consider this to be a good alignment?</li>
<li>How many unmapped reads are there? Where might this come from, and how would you determine this?</li>
<li>What could you say about library strandedness ? </li>
</ul>
<details class="success">
<summary>script and answers</summary>
<p><div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=map-multiqc</span>
<span class="c1">#SBATCH --time=00:30:00</span>
<span class="c1">#SBATCH --cpus-per-task=1</span>
<span class="c1">#SBATCH --mem=1G</span>
<span class="c1">#SBATCH -o 043_l_multiqc_map_raw.o</span>

multiqc<span class="w"> </span>-n<span class="w"> </span>043_r_multiqc_mouseMT_mapped_raw.html<span class="w"> </span>-f<span class="w"> </span>--title<span class="w"> </span>mapped_raw<span class="w"> </span>042_d_STAR_map_raw/
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/mouseMT/043_s_multiqc_map_raw.sh</code></p>
<p><a :=":" class="md-button" href="../../assets/html/043_multiqc_mouseMT_mapped_raw.html" target="_blank"> Download the report </a></p>
</details>
<h2 id="comparison-of-mapping-the-trimmed-reads">Comparison of mapping the trimmed reads</h2>
<p>After having mapped the raw reads, we also map the trimmed reads and then compare the results to decide which one we want to use for the rest of our analysis.</p>
<p>We will spare you the mapping of the trimmed reads, and let you directly download the mapping multiqc report:</p>
<p><a :=":" class="md-button" href="../../assets/html/045_multiqc_mouseMT_mapped_trimmed.html" target="_blank"> trimmed reads mapping  report </a></p>
<details class="note">
<summary>For the curious: scripts for the mapping of trimmed reads</summary>
<p><div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=star-aln</span>
<span class="c1">#SBATCH --time=00:10:00</span>
<span class="c1">#SBATCH --cpus-per-task=2</span>
<span class="c1">#SBATCH --mem=1G</span>
<span class="c1">#SBATCH -o 044_l_STAR_map_trimmed.%a.o</span>
<span class="c1">#SBATCH --array 1-8%8</span>

ml<span class="w"> </span>star

mkdir<span class="w"> </span>-p<span class="w"> </span>044_d_STAR_map_trimmed

<span class="nv">SAMPLE</span><span class="o">=</span><span class="sb">`</span>sed<span class="w"> </span>-n<span class="w"> </span><span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>p<span class="w"> </span>sampleNames.txt<span class="sb">`</span>

<span class="nv">FASTQ_NAME</span><span class="o">=</span>030_d_trim/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>.trimmed.fastq

STAR<span class="w"> </span>--runThreadN<span class="w"> </span><span class="m">4</span><span class="w"> </span>--genomeDir<span class="w"> </span>041_d_STAR_mouseMT_reference<span class="w"> </span><span class="se">\</span>
<span class="w">                 </span>--outSAMtype<span class="w"> </span>BAM<span class="w"> </span>SortedByCoordinate<span class="w"> </span><span class="se">\</span>
<span class="w">                 </span>--outFileNamePrefix<span class="w">  </span>044_d_STAR_map_trimmed/<span class="si">${</span><span class="nv">SAMPLE</span><span class="si">}</span>_trimmed.<span class="w"> </span><span class="se">\</span>
<span class="w">                 </span>--quantMode<span class="w"> </span>GeneCounts<span class="w"> </span><span class="se">\</span>
<span class="w">                 </span>--readFilesIn<span class="w"> </span><span class="nv">$FASTQ_NAME</span>
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/mouseMT/044_s_STAR_map_trimmed.sh</code></p>
<p><div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=map-trim-multiqc</span>
<span class="c1">#SBATCH --time=00:30:00</span>
<span class="c1">#SBATCH --cpus-per-task=1</span>
<span class="c1">#SBATCH --mem=1G</span>
<span class="c1">#SBATCH -o 045_l_multiqc_mouseMT_mapped_trimmed.o</span>

multiqc<span class="w"> </span>-n<span class="w"> </span>045_r_multiqc_mouseMT_mapped_trimmed.html<span class="w"> </span>-f<span class="w"> </span>--title<span class="w"> </span>mapped_trimmed<span class="w"> </span>044_d_STAR_map_trimmed/
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/mouseMT/045_s_multiqc_mouseMT_mapped_trimmed.sh</code></p>
</details>
<h2 id="additional-pseudo-aligning-with-salmon">ADDITIONAL : pseudo-aligning with salmon</h2>
<p><a :=":" class="md-button" href="https://salmon.readthedocs.io/en/latest/salmon.html" target="_blank">salmon website</a></p>
<p>Salmon can allow you to quantify transcript expression without explicitly aligning the sequenced reads onto the reference genome with its gene and splice junction annotations, but instead to a simplification of the corresponding transcriptome, thus saving computational resources.</p>
<p>We refer you to the tool&rsquo;s documentation in order to see <a href="https://salmon.readthedocs.io/en/latest/salmon.html#preparing-transcriptome-indices-mapping-based-mode">how the reference index is computed</a>.</p>
<p><strong>Task :</strong> run salmon to quantify the expression of either the Ruhland or Liu dataset. </p>
<ul>
<li>Use the tool documentation to craft your command line.</li>
<li>precomputed indices can be found in <code>/shared/data/DATA/Mouse_salmon_index</code> and <code>/shared/data/DATA/Human_salmon_index</code>.</li>
</ul>
<details class="success">
<summary>script</summary>
<p><div class="highlight"><pre><span></span><code>#!/usr/bin/bash
#SBATCH --job-name=salmonRuhland
#SBATCH --time=01:00:00
#SBATCH --cpus-per-task=8
#SBATCH --mem=30G
#SBATCH -o 033_l_salmon_ruhland2016.%a.o
#SBATCH --array 1-6%1

ml salmon

dataDIR=/shared/data/DATA/Ruhland2016

sourceFILE=Ruhland2016.fastqFiles.txt

fastqFILE=`sed -n ${SLURM_ARRAY_TASK_ID}p $sourceFILE`

genomeDIR=/shared/data/DATA/Mouse_salmon_index

outDIR=033_d_salmon_Ruhland2016_${fastqFILE%.*}

mkdir -p $outDIR

salmon quant -i $genomeDIR -l A \
            -r $dataDIR/$fastqFILE \
            -p 8 --validateMappings --gcBias --seqBias \
            -o $outDIR
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/Ruhland2016/033_s_salmon_Ruhland2016.sh</code></p>
</details>
<h2 id="additional-mapping-reads-from-ruhland2016-on-the-reference">ADDITIONAL Mapping reads from Ruhland2016 on the reference</h2>
<p><strong>Task :</strong> Using STAR, align the raw FASTQ files of the Ruhland2016 dataset against thed mouse mitochondrial reference you just created</p>
<ul>
<li>Mapping reads and generating a sorted BAM from one of the Ruhland2016 et al. FASTQ files should take about 20 minutes.</li>
<li>Use the full indexed genome at <code>/shared/data/DATA/Mouse_STAR_index/</code>, rather than the one we just made.</li>
<li><strong>IMPORTANT</strong>: use the following option in your STAR command: <code>--outTmpDir /tmp/${SLURM_JOB_USER}_${SLURM_JOB_ID}/</code>. You can use the manual to look up what this option does. The slurm variables ensure a distinct directory is created in <code>/tmp/</code> for each user and for each job.</li>
</ul>
<details class="success">
<summary>STAR mapping script of the Ruhland2016 data</summary>
<p>The following sets up an array of tasks to align all samples.</p>
<p>Source file : <code>Ruhland2016.fastqFiles.txt</code> :</p>
<div class="highlight"><pre><span></span><code>SRR3180535_EtOH1_1.fastq.gz
SRR3180536_EtOH2_1.fastq.gz
SRR3180537_EtOH3_1.fastq.gz
SRR3180538_TAM1_1.fastq.gz
SRR3180539_TAM2_1.fastq.gz
SRR3180540_TAM3_1.fastq.gz
</code></pre></div>
<p>sbatch script :</p>
<p><div class="highlight"><pre><span></span><code>#!/usr/bin/bash
#SBATCH --job-name=star-aln-Ruhland2016
#SBATCH --time=01:00:00
#SBATCH --cpus-per-task=8
#SBATCH --mem=30G
#SBATCH -o 031_l_STAR_aln_Ruhland2016.%a.o
#SBATCH --array 1-1%1


ml star
outDIR=031_d_STAR_aln_Ruhland2016

mkdir -p $outDIR

dataDIR=/shared/data/DATA/Ruhland2016

sourceFILE=Ruhland2016.fastqFiles.txt

fastqFILE=`sed -n ${SLURM_ARRAY_TASK_ID}p $sourceFILE`

genomeDIR=/shared/data/DATA/Mouse_STAR_index

STAR --runThreadN 8 --genomeDir $genomeDIR \
              --outSAMtype BAM SortedByCoordinate --outReadsUnmapped Fastx \
              --outFileNamePrefix $outDIR/$fastqFILE \
              --quantMode GeneCounts \
              --readFilesIn $dataDIR/$fastqFILE --readFilesCommand zcat \
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/mouseMT/031_s_STAR_aln_Ruhland2016.sh</code></p>
<p>The options of STAR are :</p>
<ul>
<li><strong>&ndash;runThreadN 8 </strong> : 8 threads to go faster.</li>
<li><strong>&ndash;genomeDir $genomeDIR</strong> : path of the genome to map to.</li>
<li><strong>&ndash;outSAMtype BAM SortedByCoordinate </strong> : output a coordinate-sorted BAM file.</li>
<li><strong>&ndash;outReadsUnmapped Fastx</strong> : output the non-mapping reads (in case we want to analyse them).</li>
<li><strong>&ndash;outFileNamePrefix $outDIR/$fastqFILE</strong> : prefix of output files.</li>
<li><strong>&ndash;quantMode GeneCounts</strong> : will create a file with counts of reads per gene.</li>
<li><strong>&ndash;readFilesIn $dataDIR/$fastqFILE </strong> : input read file.</li>
<li><strong>&ndash;readFilesCommand zcat</strong> : command to unzip the input file.</li>
</ul>
</details>
<h2 id="additionnal-star-2-pass">ADDITIONNAL : STAR 2-Pass</h2>
<p>Genome annotations are incomplete, particularly for complex eukaryotes : there are many as-of-yet unannotated splice junctions.</p>
<p>The first pass of STAR can create a splice junction database, containing both known and novel junctions.
This splice junction database can, in turn, be used to guide an improved second round of alignment, using a command like:</p>
<div class="highlight"><pre><span></span><code>STAR<span class="w"> </span>&lt;1st<span class="w"> </span>round<span class="w"> </span>options&gt;<span class="w"> </span>--sjdbFileChrStartEnd<span class="w"> </span>sample_SJ.out.tab
</code></pre></div>
<p><strong>Task :</strong> run STAR in this STAR-2pass mode on the same sample as before and evaluate the results.</p>
<details class="success">
<summary>script</summary>
<p><div class="highlight"><pre><span></span><code>#!/usr/bin/bash
#SBATCH --job-name=star-aln2-Ruhland2016
#SBATCH --time=01:00:00
#SBATCH --cpus-per-task=8
#SBATCH --mem=30G
#SBATCH -o 032_l_STAR_2PASS_Ruhland2016.%a.o
#SBATCH --array 1-1%1

ml star

outDIR=032_d_STAR_Ruhland2016

mkdir -p $outDIR

dataDIR=/shared/data/DATA/Ruhland2016

sourceFILE=Ruhland2016.fastqFiles.txt

fastqFILE=`sed -n ${SLURM_ARRAY_TASK_ID}p $sourceFILE`

genomeDIR=/shared/data/DATA/Mouse_STAR_index

STAR --runThreadN 8 --genomeDir $genomeDIR \
                  --outSAMtype BAM SortedByCoordinate \
                  --outFileNamePrefix $outDIR/$fastqFILE.2Pass. \
                  --outReadsUnmapped Fastx --quantMode GeneCounts \
                  --sjdbFileChrStartEnd $outDIR/${fastqFILE}SJ.out.tab \
                  --readFilesIn $dataDIR/$fastqFILE --readFilesCommand zcat \
</code></pre></div>
it can also be found in the cluster at <code>/shared/data/Solutions/mouseMT/032_s_STAR_2PASS_Ruhland2016.sh</code></p>
</details>
<!--
## ADDITIONNAL : Assessing read coverage for biases

The [RSeQC](http://rseqc.sourceforge.net/) package includes a function for evaluating [“gene body coverage”](http://rseqc.sourceforge.net/#genebody-coverage-py), which
can be used to assess 5’ or 3’ bias, which might happen if your RNA is degraded or otherwise biased

Requirements:

 * Genome annotations in the 12-column BED format : [you can grab one for mouse here](https://sourceforge.net/projects/rseqc/files/BED/Mouse_Mus_musculus/), or at `/data/GRCm38/Mus_musculus.GRCm38.89.bed12` on the server `CHECK LINK. gtf2bed instead?`
 * *Indexed* and sorted BAM file, which can be generated from a sorted BAM using the SAMtools package: `samtools index sample1_sorted.bam`

Example command :
<div class="highlight"><pre><span></span><code>geneBody_coverage.py<span class="w"> </span>-r<span class="w"> </span>/data/GRCm38/Mus_musculus.GRCm38.89.bed12<span class="w"> </span><span class="se">\</span>
<span class="w">                     </span>-i<span class="w"> </span>sample1_sorted.bam<span class="w"> </span><span class="se">\</span>
<span class="w">                     </span>-f<span class="w"> </span>pdf<span class="w"> </span><span class="se">\</span>
<span class="w">                     </span>-o<span class="w"> </span>output_prefix
</code></pre></div>

**Task** : Evaluate the gene body coverage for the sample you aligned.

 * expected RAM : 
 * expected time : 

??? done "geneBody_coverage script"

    TBD 
-->





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.74e28a9f.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.220ee61c.min.js"></script>
      
    
  </body>
</html>