
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      <link rel="icon" href="../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.2.1, mkdocs-material-7.1.8">
    
    
      
        <title>Server login + unix fresh up - Introduction to RNA-Seq - From quality control to pathway analysis</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.ca7ac06f.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.f1a3b89f.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    
      


    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#the-computing-cluster" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../index.html" title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-header__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to RNA-Seq - From quality control to pathway analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Server login + unix fresh up
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/sib-swiss/RNAseq-introduction-training/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../index.html" title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-nav__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to RNA-Seq - From quality control to pathway analysis
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/sib-swiss/RNAseq-introduction-training/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../index.html" class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../precourse.html" class="md-nav__link">
        Precourse preparations
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../course_schedule.html" class="md-nav__link">
        Course schedule
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" checked>
      
      <label class="md-nav__link" for="__nav_4">
        Day1
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Day1" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Day1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="design.html" class="md-nav__link">
        RNAseq - technologies and design
      </a>
    </li>
  

          
            
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Server login + unix fresh up
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="server_login.html" class="md-nav__link md-nav__link--active">
        Server login + unix fresh up
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#the-computing-cluster" class="md-nav__link">
    The computing cluster
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#connect-to-the-server" class="md-nav__link">
    Connect to the server
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-command-line-on-the-cluster" class="md-nav__link">
    Using command line on the cluster
  </a>
  
    <nav class="md-nav" aria-label="Using command line on the cluster">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#creating-a-directory" class="md-nav__link">
    Creating a directory
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#creating-and-editing-a-file" class="md-nav__link">
    Creating and editing a file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exchanging-files-with-the-server" class="md-nav__link">
    Exchanging files with the server
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#submitting-jobs" class="md-nav__link">
    Submitting jobs
  </a>
  
    <nav class="md-nav" aria-label="Submitting jobs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#advanced-cluster-usage-loading-modules" class="md-nav__link">
    Advanced cluster usage : loading modules
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#advanced-cluster-usage-job-array" class="md-nav__link">
    Advanced cluster usage : job array
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="quality_control.html" class="md-nav__link">
        Quality control
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="trimming.html" class="md-nav__link">
        Sequence trimming
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="mapping.html" class="md-nav__link">
        Reads mapping
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      <label class="md-nav__link" for="__nav_5">
        Day2
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Day2" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Day2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="counting.html" class="md-nav__link">
        Read counting
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="DE.html" class="md-nav__link">
        Differential Expression Inference
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#the-computing-cluster" class="md-nav__link">
    The computing cluster
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#connect-to-the-server" class="md-nav__link">
    Connect to the server
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-command-line-on-the-cluster" class="md-nav__link">
    Using command line on the cluster
  </a>
  
    <nav class="md-nav" aria-label="Using command line on the cluster">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#creating-a-directory" class="md-nav__link">
    Creating a directory
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#creating-and-editing-a-file" class="md-nav__link">
    Creating and editing a file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#exchanging-files-with-the-server" class="md-nav__link">
    Exchanging files with the server
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#submitting-jobs" class="md-nav__link">
    Submitting jobs
  </a>
  
    <nav class="md-nav" aria-label="Submitting jobs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#advanced-cluster-usage-loading-modules" class="md-nav__link">
    Advanced cluster usage : loading modules
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#advanced-cluster-usage-job-array" class="md-nav__link">
    Advanced cluster usage : job array
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/sib-swiss/RNAseq-introduction-training/edit/master/docs/days/server_login.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  <h1>Server login + unix fresh up</h1>
                
                <p>To conduct the practicals of this course, we will be using a dedicated High Performance Computing cluster. 
This matches the reality of most NGS workflows, which cannot be completed in a reasonable time on a single machine. </p>
<p>To interact with this cluster, you will have to log in to a distant <em>head node</em>. From there you will be able to distribute your computational tasks to the cluster using a <em>job scheduler</em> called Slurm.</p>
<p>This page will cover our first contact with the distant cluster. </p>
<p><strong>You will learn to :</strong></p>
<ul>
<li>understand a typical computer cluster architecture.</li>
<li>connect to the server.</li>
<li>use the command line to perform basic operations on the head node.</li>
<li>exchange files between the server and your own machine.</li>
<li>submit a job to the cluster.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you are doing this course on your own, then the distant server provided within the course will not be available. 
Feel free to ignore or adapt any of the following steps to your own situation.</p>
</div>
<h2 id="the-computing-cluster">The computing cluster</h2>
<p>The computing cluster follows an architecture that enables several users to distribute computational tasks among several machines which share a number of resources, such as a common file system.</p>
<p><img alt="cluster_overview" src="../assets/images/cluster_overview.png" /></p>
<p>Users do not access each machine individually, but rather connect to a <strong>head node</strong>. From there, they can interact with the cluster using the <strong>job scheduler</strong> (here slurm).
The job scheduler&rsquo;s role is to manage where and how to run the jobs of all users, such that waiting time is minimized and resource usage is optimized.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Everyone is connected to the same head node. Do not perform compute-intensive tasks on it or you will slow everyone down! </p>
</div>
<h2 id="connect-to-the-server">Connect to the server</h2>
<p>Say you want to connect to cluster with address <code>xx.xx.xx.xx</code> and your login is <code>login</code>.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you are doing this course with a teacher, use the link, login and password provided before or during the course. </p>
</div>
<p>The first step will be to open a <strong>terminal</strong></p>
<div class="tabbed-set" data-tabs="1:3"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><label for="__tabbed_1_1">Mac</label><div class="tabbed-content">
<p>Open a terminal, for instance with the application Xterm, or Xquartz.
<!-- Might confuse some who just read "Terminal" --></p>
</div>
<input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><label for="__tabbed_1_2">Linux</label><div class="tabbed-content">
<p>Open a new terminal.</p>
</div>
<input id="__tabbed_1_3" name="__tabbed_1" type="radio" /><label for="__tabbed_1_3">Windows</label><div class="tabbed-content">
<p>Open the application mobaXterm (or any ssh-enabling terminal aplpication you prefer).</p>
<p>On mobaXterm, click on &ldquo;Start a local Terminal&rdquo;.</p>
</div>
</div>
<hr />
<p>In the terminal type the following command:</p>
<div class="highlight"><pre><span></span><code>ssh login@xx.xx.xx.xx
</code></pre></div>
<p>When prompted for your password, type it and press Enter. </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
</div>
<p>There is no cursor or &lsquo;‚óè&rsquo; character appearing while you type your password. This is normal.</p>
<p>After a few seconds, you should be logged into the <em>head node</em> and ready to begin.</p>
<h2 id="using-command-line-on-the-cluster">Using command line on the cluster</h2>
<p>Now that you are in the head node, it is time to get acquainted with your environment and to prepare the upcoming practicals. 
We will also use this as a short reminder about the UNIX command line.</p>
<p>You can also refer to this nice <a href="https://appletree.or.kr/quick_reference_cards/Unix-Linux/Linux%20Command%20Line%20Cheat%20Sheet.pdf">Linux Command Line Cheat Sheet</a>.</p>
<hr />
<p>At any time, you can get the location (folder) your terminal is in at by typing the &ldquo;print working directory&rdquo; command:</p>
<div class="highlight"><pre><span></span><code><span class="nb">pwd</span>
</code></pre></div>
<p>When you start a session on a distant computer, you are placed in your <code>home</code> directory. So the cluster should return something like:</p>
<div class="highlight"><pre><span></span><code>/shared/home/&lt;login&gt;
</code></pre></div>
<h3 id="creating-a-directory">Creating a directory</h3>
<p>Use the command line to create a repository called <code>day1</code> where you will put all materials relating to this first day.</p>
<details class="done"><summary>Answer</summary><div class="highlight"><pre><span></span><code>mkdir day1
</code></pre></div>
</details>
<p>Move to that directory.</p>
<details class="done"><summary>Answer</summary><div class="highlight"><pre><span></span><code><span class="nb">cd</span> day1
</code></pre></div>
</details>
<p>The directory <code>/shared/data/</code> contains data and solutions for most practicals. Check the content of that directory.</p>
<details class="done"><summary>Answer</summary><div class="highlight"><pre><span></span><code>ls /shared/data/
</code></pre></div>
</details>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You don&rsquo;t need to move to that directory to list its contents!</p>
</div>
<p>Copy the script <code>fastqc_Liu2015_SRR1272187_1.sh</code> from  <code>/shared/data/Solutions/Liu2015</code> into your current directory.</p>
<details class="done"><summary>Answer</summary><div class="highlight"><pre><span></span><code>cp /shared/data/Solutions/Liu2015/fastqc_Liu2015_SRR1272187_1.sh .
</code></pre></div>
</details>
<p>Print the content of this script to the screen.</p>
<details class="done"><summary>Answer</summary><p><div class="highlight"><pre><span></span><code>more fastqc_Liu2015_SRR1272187_1.sh
</code></pre></div>
output:
<div class="highlight"><pre><span></span><code>#!/usr/bin/bash
#SBATCH --job-name=fastqc
#SBATCH --time=00:30:00
#SBATCH --cpus-per-task=1
#SBATCH --mem=1G
#SBATCH -o fastqc_Liu2015.o
#SBATCH -e fastqc_Liu2015.e

ml fastqc

dataDir=/shared/data/DATA/Liu2015

mkdir -p FASTQC_Liu2015/
fastqc -o FASTQC_Liu2015/ $dataDir/SRR1272187_1.fastq.gz
</code></pre></div></p>
</details>
<p>We&rsquo;ll see what all this means soon.</p>
<h3 id="creating-and-editing-a-file">Creating and editing a file</h3>
<p>To edit files on the distant server, we will use the command line editor <code>nano</code>. It is far from the most complete or efficient one, but it can be found on most servers and is arguably among the easiest to start with.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Alternatively, feel free to use any other CLI editor you prefer, such as <code>vi</code>.</p>
</div>
<p>To start editing a file named <code>test.txt</code>, type :</p>
<div class="highlight"><pre><span></span><code>    nano test.txt
</code></pre></div>
<p>You will be taken to the <code>nano</code> interface :</p>
<p><img alt="nano screenshot" src="../assets/images/nano_screenshot.png" /></p>
<p>Type in your favorite movie quote, and then exit by pressing <code>Ctrl+x</code> (<code>control+x</code> on a Mac keyboard), and then <code>y</code> and <code>Enter</code> when prompted to save the modifications you just made.</p>
<p>You can check that your modifications were saved by typing</p>
<div class="highlight"><pre><span></span><code>more test.txt
</code></pre></div>
<h3 id="exchanging-files-with-the-server">Exchanging files with the server</h3>
<p>Whether you want to transfer some data to the cluster or retrieve the results of your latest computation, it is important to be able to exchange files with the distant server.</p>
<p>There exists several alternatives, depending on your platform and preferences.</p>
<div class="tabbed-set" data-tabs="2:3"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><label for="__tabbed_2_1">command line</label><div class="tabbed-content">
<p>We will use <code>scp</code>.</p>
<p>To copy a file from the server to your machine, use this syntax on a <strong>terminal in your local machine</strong> (open a new terminal if necessary).</p>
<div class="highlight"><pre><span></span><code>scp &lt;login&gt;@&lt;server-adress&gt;:/path/to/file/on/server/file.txt /local/destination/
</code></pre></div>
<p>For example, to copy the file <code>test.txt</code> you just created in the folder <code>day1/</code>, to your current (local) working directory :</p>
<div class="highlight"><pre><span></span><code>scp login@xx.xx.xx.xx:~/day1/file.txt .
</code></pre></div>
<p>To copy a file from your machine to the server (NB: here <code>~</code> will be interpreted as your home directory, this is a useful and time-saving shorthand):</p>
<div class="highlight"><pre><span></span><code>scp /path/to/file/local/file.txt &lt;login&gt;@&lt;server-adress&gt;:/destination/on/server/
</code></pre></div>
</div>
<input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><label for="__tabbed_2_2">graphical alternative</label><div class="tabbed-content">
<p>There are nice and free software with graphical user interfaces, such as <a href="https://filezilla-project.org/">filezilla</a>, to help you manage exchanges with the distant server. Feel free to install and experiment with it during the break.
<img alt="filezilla_mainscreen" src="https://filezilla-project.org/images/screenshots/fz3_win_main.png" /></p>
</div>
<input id="__tabbed_2_3" name="__tabbed_2" type="radio" /><label for="__tabbed_2_3">mobaXterm</label><div class="tabbed-content">
<p>If you are using mobaXterm, the left panel should provide a graphical SFTP browser in the left sidebar which allows you to browse and drag and drop files directly from/to the remote server.</p>
<p><img alt="mobaxterm_leftpanel" src="https://mobaxterm.mobatek.net/img/moba/features/feature-sftp-browser.png" /></p>
</div>
</div>
<h2 id="submitting-jobs">Submitting jobs</h2>
<p>Jobs can be submitted to the compute cluster using <strong>sbatch scripts</strong>, which contain 2 parts :</p>
<ul>
<li>
<p>information for the job scheduler:</p>
<ul>
<li>how much RAM / CPUs do I need ?</li>
<li>where to write the logs of my job ?</li>
</ul>
</li>
<li>
<p>bash commands corresponding to your task</p>
</li>
</ul>
<p>But an example is worth a thousand words :</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=test</span>
<span class="c1">#SBATCH --time=00:30:00</span>
<span class="c1">#SBATCH --cpus-per-task=1</span>
<span class="c1">#SBATCH --mem=1G</span>
<span class="c1">#SBATCH -o test_log.o</span>


<span class="nb">echo</span> <span class="s2">&quot;looking at the size of the elements of /shared/data/&quot;</span>
sleep <span class="m">10</span> <span class="c1"># making the script wait for 10 seconds - this is just so we can see it later on. </span>
<span class="c1"># `du` is &quot;disk usage&quot;, a command that returns the size of a folder structure.</span>
du -h -d <span class="m">2</span> /shared/data/
</code></pre></div>
<p>The lines beginning by <code>#SBATCH</code> specify options to the job scheduler:</p>
<ul>
<li>The first line indicates what shell should be used to interpret the commands.</li>
<li><code>#SBATCH --job-name=test</code> : the job name</li>
<li><code>#SBATCH --time=00:30:00</code> : time reserved for the job : 30min. </li>
<li><code>#SBATCH --cpus-per-task=1</code> : CPUs for the job </li>
<li><code>#SBATCH --mem=1G</code> : memory for the job</li>
<li><code>#SBATCH -o test_log.o</code> : file to write output or error messages</li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Your job will fail as soon as it takes more time or RAM than requested. You might need to test it to find the appropriate values.</p>
</div>
<p>Copy this script inside a new file named <code>mySbatchScript.sh</code>, then submit it to the job scheduler using :</p>
<div class="highlight"><pre><span></span><code>    sbatch mySbatchScript.sh
</code></pre></div>
<p>Afterward, use the command <code>squeue</code> to monitor the jobs submitted to the cluster. Locate your job and wait for it to be accepted (<code>RUNNING</code> status), and then to complete (the job disappears from the output of <code>squeue</code>). </p>
<p>Check the output of your job in the output file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When there are a lot of jobs, <code>squeue -u &lt;username&gt;</code> will limit the list to those of the specified user.</p>
</div>
<h3 id="advanced-cluster-usage-loading-modules">Advanced cluster usage : loading modules</h3>
<p>During our various analysis, we will call onto numerous software.</p>
<p>Fortunately, in most case we do not have to install each of these ourselves onto the cluster : they have already been packaged and prepared to be made available to you or your code.</p>
<p>However, by default these are not loaded and you have to explicitely load the module containing the software you want in your script (or in the interactive shell session).</p>
<p><strong>Question:</strong> Why aren&rsquo;t all the module already pre-loaded ?</p>
<details class="done"><summary>Answer</summary><p>Many toolsets have dependencies toward different, sometimes incompatible libraries. Packaging each tool independently and loading them separately circumvents this as you only load what you need, and you can always unload a toolset if you need to load another, incompatible, toolset.</p>
</details>
<p>Modules are managed with the <code>module</code> command.</p>
<p>Basic commands are :</p>
<ul>
<li><code>module list</code> : lists currently loaded modules</li>
<li><code>module load &lt;modulename&gt;</code> alias <code>ml &lt;modulename&gt;</code> : loads module <code>&lt;modulename&gt;</code></li>
<li><code>module unload &lt;modulename&gt;</code>  : unloads module <code>&lt;modulename&gt;</code></li>
<li><code>module purge</code> : unloads all loaded modules</li>
<li><code>module avail</code> : lists all modules available for loading</li>
<li><code>module keyword &lt;KW&gt;</code> : lists all modules available for loading which contains <code>&lt;KW&gt;</code></li>
</ul>
<p>Try it for yourself: soon, we will need the fastqc software. 
If we type in the terminal:
<div class="highlight"><pre><span></span><code>fastqc
</code></pre></div>
this should give you an error, telling you there is no such command.</p>
<p>Now, if we load it first
<div class="highlight"><pre><span></span><code>ml fastqc # shortcut for &quot;module load fastqc&quot;
fastqc
</code></pre></div>
Now you should not have any error.</p>
<blockquote>
<p>Note: our module provider is ComputeCanada, which has a lot of <a href="https://docs.alliancecan.ca/wiki/Available_software">available software</a>. To avoid storing all these on our cluster, each time a new module is loaded, it is fetched first on the compute canada servers, so sometimes it can take a bit of time to load a module for the first time.</p>
</blockquote>
<h3 id="advanced-cluster-usage-job-array">Advanced cluster usage : job array</h3>
<p>Often, we have to repeat a similar analysis on a number of files, or for a number of different parameters.
Rather than writing each sbatch script individually, we can rely on job arrays to facilitate our task.</p>
<p>Say you want to execute a command, on 10 files (for example, map the reads of 10 samples).</p>
<p>You first create a file containing the name of your files (one per line); let&rsquo;s call it <code>readFiles.txt</code>.</p>
<p>Then, you write an sbatch array job script:</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>
<span class="c1">#SBATCH --job-name=test_array</span>
<span class="c1">#SBATCH --time=00:30:00</span>
<span class="c1">#SBATCH --cpus-per-task=1</span>
<span class="c1">#SBATCH --mem=1G</span>
<span class="c1">#SBATCH -o test_array_log.%a.o</span>
<span class="c1">#SBATCH --array 1-10%5</span>


<span class="nb">echo</span> <span class="s2">&quot;job array id&quot;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>

<span class="c1"># sed -n &lt;X&gt;p &lt;file&gt; : retrieve line &lt;X&gt; of file</span>
<span class="c1"># so the next line grabs the file name corresponding to our job array task id and stores it in the variable ReadFileName </span>
<span class="nv">ReadFileName</span><span class="o">=</span><span class="sb">`</span>sed -n <span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>p readFiles.txt<span class="sb">`</span>

<span class="c1"># here we would put the mapping command or whatever</span>
<span class="nb">echo</span> <span class="nv">$ReadFileName</span>
</code></pre></div>
<p>Some things have changed compared to the previous sbatch script :</p>
<ul>
<li><code>#SBATCH --array 1-10%5</code> : will spawn independent tasks with ids from 1 to 10, and will manage them so that at most 5 run at the same time.</li>
<li><code>#SBATCH -o test_array_log.%a.o</code> : the <code>%a</code> will take the value of the array task id. So we will have 1 log file per task (so 10 files).</li>
<li><code>$SLURM_ARRAY_TASK_ID</code> : changes value between the different tasks. This is what we use to execute the same script on different files (using <code>sed -n ${SLURM_ARRAY_TASK_ID}p</code>)</li>
</ul>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="design.html" class="md-footer__link md-footer__link--prev" aria-label="Previous: RNAseq - technologies and design" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              RNAseq - technologies and design
            </div>
          </div>
        </a>
      
      
        
        <a href="quality_control.html" class="md-footer__link md-footer__link--next" aria-label="Next: Quality control" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Quality control
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../assets/javascripts/workers/search.b0710199.min.js", "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.76f349be.min.js"></script>
      
    
  </body>
</html>