
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../mapping/">
      
      
        <link rel="next" href="../enrichment/">
      
      <link rel="icon" href="../../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.18">
    
    
      
        <title>Differential Expression Inference - Introduction to RNA-Seq - From quality control to pathway analysis</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.26e3688c.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#material" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-header__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to RNA-Seq - From quality control to pathway analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Differential Expression Inference
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/RNAseq-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Introduction to RNA-Seq - From quality control to pathway analysis" class="md-nav__button md-logo" aria-label="Introduction to RNA-Seq - From quality control to pathway analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to RNA-Seq - From quality control to pathway analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/RNAseq-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/RNAseq-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../precourse/" class="md-nav__link">
        Precourse preparations
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../course_schedule/" class="md-nav__link">
        Course schedule
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../slides_notes/" class="md-nav__link">
        Slides notes (draft)
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../cheatsheets/" class="md-nav__link">
        Cheatsheets
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
      
      
      
        <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Day1
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../design/" class="md-nav__link">
        RNAseq - technologies and design
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../server_login/" class="md-nav__link">
        Server login + unix fresh up
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../quality_control/" class="md-nav__link">
        Quality control
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../trimming/" class="md-nav__link">
        Sequence trimming
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../mapping/" class="md-nav__link">
        Reads mapping
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" checked>
      
      
      
        <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Day2
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Differential Expression Inference
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Differential Expression Inference
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#connexion-to-the-rstudio-server" class="md-nav__link">
    Connexion to the Rstudio server
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-expression-inference" class="md-nav__link">
    Differential Expression Inference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reading" class="md-nav__link">
    reading
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#setting-up-the-experimental-design" class="md-nav__link">
    setting up the experimental design
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-the-deseq-data-object-and-some-qc" class="md-nav__link">
    creating the DESeq data object and some QC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-without-the-outliers" class="md-nav__link">
    analysis without the outliers
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#saving-results-to-file" class="md-nav__link">
    saving results to file
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reading_1" class="md-nav__link">
    reading
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#experimental-design" class="md-nav__link">
    experimental design
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#edger-object-preprocessing-and-qc" class="md-nav__link">
    edgeR  object preprocessing and QC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-without-the-outliers_1" class="md-nav__link">
    analysis without the outliers
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#writing-the-table-of-results" class="md-nav__link">
    writing the table of results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-expression-task" class="md-nav__link">
    Differential Expression - Task
  </a>
  
    <nav class="md-nav" aria-label="Differential Expression - Task">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ruhland2016-deseq2-correction" class="md-nav__link">
    Ruhland2016 - DESeq2 correction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ruhland2016-edger-correction" class="md-nav__link">
    Ruhland2016 - EdgeR correction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tuch-2010-edger-correction" class="md-nav__link">
    Tuch 2010 - EdgeR correction
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-importing-counts-from-salmon-with-tximport" class="md-nav__link">
    Additional : importing counts from salmon with tximport
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../enrichment/" class="md-nav__link">
        Enrichment analysis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#connexion-to-the-rstudio-server" class="md-nav__link">
    Connexion to the Rstudio server
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-expression-inference" class="md-nav__link">
    Differential Expression Inference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reading" class="md-nav__link">
    reading
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#setting-up-the-experimental-design" class="md-nav__link">
    setting up the experimental design
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-the-deseq-data-object-and-some-qc" class="md-nav__link">
    creating the DESeq data object and some QC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-without-the-outliers" class="md-nav__link">
    analysis without the outliers
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#saving-results-to-file" class="md-nav__link">
    saving results to file
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reading_1" class="md-nav__link">
    reading
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#experimental-design" class="md-nav__link">
    experimental design
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#edger-object-preprocessing-and-qc" class="md-nav__link">
    edgeR  object preprocessing and QC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-without-the-outliers_1" class="md-nav__link">
    analysis without the outliers
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#writing-the-table-of-results" class="md-nav__link">
    writing the table of results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-expression-task" class="md-nav__link">
    Differential Expression - Task
  </a>
  
    <nav class="md-nav" aria-label="Differential Expression - Task">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ruhland2016-deseq2-correction" class="md-nav__link">
    Ruhland2016 - DESeq2 correction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ruhland2016-edger-correction" class="md-nav__link">
    Ruhland2016 - EdgeR correction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tuch-2010-edger-correction" class="md-nav__link">
    Tuch 2010 - EdgeR correction
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-importing-counts-from-salmon-with-tximport" class="md-nav__link">
    Additional : importing counts from salmon with tximport
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


  <h1>Differential Expression Inference</h1>

<p>Once the reads have been mapped and counted, one can assess the differential expression of genes between different conditions.</p>
<p><strong>During this lesson, you will learn to :</strong></p>
<ul>
<li>describe the different steps of data normalization and modelling commonly used for RNA-seq data.</li>
<li>detect significantly differentially-expressed genes using either edgeR or DESeq2.</li>
</ul>
<h2 id="material">Material</h2>
<p><a :=":" class="md-button" href="../../assets/pdf/RNA-Seq_06_DE.pdf" target="_blank"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0h160v128c0 17.7 14.3 32 32 32h128v144H176c-35.3 0-64 28.7-64 64v144H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0l128 128zM176 352h32c30.9 0 56 25.1 56 56s-25.1 56-56 56h-16v32c0 8.8-7.2 16-16 16s-16-7.2-16-16V368c0-8.8 7.2-16 16-16zm32 80c13.3 0 24-10.7 24-24s-10.7-24-24-24h-16v48h16zm96-80h32c26.5 0 48 21.5 48 48v64c0 26.5-21.5 48-48 48h-32c-8.8 0-16-7.2-16-16V368c0-8.8 7.2-16 16-16zm32 128c8.8 0 16-7.2 16-16v-64c0-8.8-7.2-16-16-16h-16v96h16zm80-112c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16s-7.2 16-16 16h-32v32h32c8.8 0 16 7.2 16 16s-7.2 16-16 16h-32v48c0 8.8-7.2 16-16 16s-16-7.2-16-16V368z"/></svg></span> Download the presentation</a></p>
<p><a href="https://www.rstudio.com/">Rstudio website</a></p>
<!-- Suggestion: RStudio reminders ??? Or link to some course or something? -->
<!-- Perhaps this one? https://www.datacamp.com/tutorial/r-studio-tutorial -->

<p><a :=":" class="md-button" href="https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf" target="_blank">edgeR user&rsquo;s guide</a></p>
<p><a :=":" class="md-button" href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html" target="_blank">DESeq2 vignette</a></p>
<h2 id="connexion-to-the-rstudio-server">Connexion to the Rstudio server</h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This step is intended only for users who attend the course with a teacher. Otherwise you will have to rely on your own installation of Rstudio.</p>
</div>
<p>The analysis of the read count data will be done on an RStudio instance, using the R language and some relevant <a href="http://bioconductor.org/">Bioconductor</a> libraries.</p>
<p>As you start your session on the RStudio server, please make sure that you know where your data is situated with respect to your <strong>working directory</strong> (use <code>getwd()</code> and <code>setwd()</code> to respectively : know what your working directory is, and change it as necessary).</p>
<h2 id="differential-expression-inference">Differential Expression Inference</h2>
<p>Let&rsquo;s analyze the <code>mouseMT</code> toy dataset.</p>
<details class="note">
<summary>DESeq2 analysis</summary>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">DESeq2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">pheatmap</span><span class="p">)</span>
</code></pre></div>
<h2 id="reading">reading</h2>
<p>We will use the trimmed reads</p>
<div class="highlight"><pre><span></span><code><span class="n">folder</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/shared/data/Solutions/mouseMT/044_STAR_map_trimmed/&quot;</span>

<span class="c1"># we skip the 4 first lines, which contains </span>
<span class="c1"># N_unmapped , N_multimapping , N_noFeature , N_ambiguous   </span>

<span class="n">sample_a1_table</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="w"> </span><span class="n">folder</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sample_a1&quot;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.ReadsPerGene.out.tab&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">skip</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="w"> </span><span class="n">sample_a1_table</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>                    V2      V3      V4
ENSMUSG00000064336  0       0       0   
ENSMUSG00000064337  0       0       0   
ENSMUSG00000064338  0       0       0   
ENSMUSG00000064339  0       0       0   
ENSMUSG00000064340  0       0       0   
ENSMUSG00000064341  4046    1991    2055    
</code></pre></div>
<p>We are interested in the first columns, which contains counts for unstranded reads</p>
<p>Let&rsquo;s use a loop to automatize the reading:
<div class="highlight"><pre><span></span><code><span class="n">raw_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="nf">row.names</span><span class="p">(</span><span class="n">sample_a1_table</span><span class="p">)</span><span class="w"> </span><span class="p">)</span>

<span class="nf">for</span><span class="p">(</span><span class="w"> </span><span class="n">sample</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;a1&#39;</span><span class="p">,</span><span class="s">&#39;a2&#39;</span><span class="p">,</span><span class="s">&#39;a3&#39;</span><span class="p">,</span><span class="s">&#39;a4&#39;</span><span class="p">,</span><span class="s">&#39;b1&#39;</span><span class="p">,</span><span class="s">&#39;b2&#39;</span><span class="p">,</span><span class="s">&#39;b3&#39;</span><span class="p">,</span><span class="s">&#39;b4&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">){</span>
<span class="w">  </span><span class="n">sample_table</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="w"> </span><span class="n">folder</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sample_&quot;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">sample</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.ReadsPerGene.out.tab&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span>
<span class="w">                            </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">skip</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="p">)</span>

<span class="w">  </span><span class="n">raw_counts</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sample_table</span><span class="p">[</span><span class="w"> </span><span class="nf">row.names</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;V2&quot;</span><span class="w"> </span><span class="p">]</span>

<span class="p">}</span>

<span class="nf">head</span><span class="p">(</span><span class="w"> </span><span class="n">raw_counts</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>                    a1      a2      a3      a4  b1  b2  b3  b4
ENSMUSG00000064336  0       0       0       0   0   0   0   0
ENSMUSG00000064337  0       0       0       0   0   0   0   0
ENSMUSG00000064338  0       0       0       0   0   0   0   0
ENSMUSG00000064339  0       0       0       2   0   0   0   0
ENSMUSG00000064340  0       0       0       0   0   0   0   0
ENSMUSG00000064341  4046    4098    4031    1   449 515 13  456
</code></pre></div></p>
<h2 id="setting-up-the-experimental-design">setting up the experimental design</h2>
<div class="highlight"><pre><span></span><code><span class="c1">#note: levels let&#39;s us define the reference levels</span>
<span class="n">treatment</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;a&quot;</span><span class="p">,</span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;b&quot;</span><span class="p">,</span><span class="m">4</span><span class="p">)),</span><span class="w"> </span><span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;a&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;b&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">)</span>
<span class="n">colData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">treatment</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">))</span>
<span class="n">colData</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>    treatment
a1  a           
a2  a           
a3  a           
a4  a           
b1  b           
b2  b           
b3  b           
b4  b
</code></pre></div>
<h2 id="creating-the-deseq-data-object-and-some-qc">creating the DESeq data object and some QC</h2>
<p><div class="highlight"><pre><span></span><code><span class="n">dds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DESeqDataSetFromMatrix</span><span class="p">(</span>
<span class="w">  </span><span class="n">countData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="p">,</span><span class="w"> </span><span class="n">colData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colData</span><span class="p">,</span><span class="w"> </span>
<span class="w">  </span><span class="n">design</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 37  8
</code></pre></div></p>
<p>Filter low count genes. </p>
<p>Here, we will apply a very soft filter and keep genes with at least 1 read in at least 4 samples (size of the smallest group).</p>
<p><div class="highlight"><pre><span></span><code><span class="n">idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowSums</span><span class="p">(</span><span class="nf">counts</span><span class="p">(</span><span class="n">dds</span><span class="p">,</span><span class="w"> </span><span class="n">normalized</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">4</span>
<span class="n">dds.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dds</span><span class="p">[</span><span class="n">idx</span><span class="p">,</span><span class="w"> </span><span class="p">]</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 13  8
</code></pre></div></p>
<p>We go from 37 to 13 genes</p>
<p>We perform the estimation of dispersions 
<div class="highlight"><pre><span></span><code><span class="n">dds.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DESeq</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
-- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
   function: y = a/x + b, and a local regression fit was automatically substituted.
   specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.
final dispersion estimates
fitting model and testing
</code></pre></div></p>
<p>PCA plot of the samples:
<div class="highlight"><pre><span></span><code><span class="c1"># blind : whether to blind the transformation to the experimental design. </span>
<span class="c1">#   - blind=TRUE : comparing samples in a manner unbiased by prior information on samples, </span>
<span class="c1">#                  for example to perform sample QA (quality assurance).</span>
<span class="c1">#   - blind=FALSE: should be used for transforming data for downstream analysis, </span>
<span class="c1">#                  where the full use of the design information should be made.</span>

<span class="n">vsd</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">varianceStabilizingTransformation</span><span class="p">(</span><span class="n">dds.f</span><span class="p">,</span><span class="w"> </span><span class="n">blind</span><span class="o">=</span><span class="kc">TRUE</span><span class="w"> </span><span class="p">)</span>
<span class="n">pcaData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">plotPCA</span><span class="p">(</span><span class="n">vsd</span><span class="p">,</span><span class="w"> </span><span class="n">intgroup</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;treatment&quot;</span><span class="p">))</span>
<span class="n">pcaData</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_label</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PC1</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">PC2</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">))</span>
</code></pre></div></p>
<p><img alt="pca all samples" src="../../assets/images/DESeq2_mouseMT/mouseMT_pca1.png" /></p>
<p>OK, so a4 and b3 are quite different from the rest.</p>
<ul>
<li>a4 was expected from the QC</li>
<li>b3 we did not expect until now</li>
</ul>
<p>If we did the analysis with them, here is what we get:
<div class="highlight"><pre><span></span><code><span class="n">res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">results</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>out of 13 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 0, 0%
LFC &lt; 0 (down)     : 0, 0%
outliers [1]       : 7, 54%
low counts [2]     : 0, 0%
(mean count &lt; 16)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results
</code></pre></div></p>
<p>So, let&rsquo;s eliminate these two samples.</p>
<h2 id="analysis-without-the-outliers">analysis without the outliers</h2>
<div class="highlight"><pre><span></span><code><span class="n">raw_counts_no_outliers</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="p">[</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="o">!</span><span class="p">(</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;a4&#39;</span><span class="p">,</span><span class="s">&#39;b3&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">]</span>

<span class="n">treatment</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;a&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;b&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">)),</span><span class="w"> </span><span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;a&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;b&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">)</span>
<span class="n">colData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">treatment</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">raw_counts_no_outliers</span><span class="p">))</span>
<span class="n">colData</span><span class="w"> </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>    treatment
a1  a           
a2  a           
a3  a           
b1  b           
b2  b           
b4  b
</code></pre></div>
<p><div class="highlight"><pre><span></span><code><span class="n">dds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DESeqDataSetFromMatrix</span><span class="p">(</span>
<span class="w">  </span><span class="n">countData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts_no_outliers</span><span class="p">,</span><span class="w"> </span><span class="n">colData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colData</span><span class="p">,</span><span class="w"> </span>
<span class="w">  </span><span class="n">design</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 37  6
</code></pre></div></p>
<p>Filter low count genes: now the smallest group is 3
<div class="highlight"><pre><span></span><code><span class="n">idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowSums</span><span class="p">(</span><span class="nf">counts</span><span class="p">(</span><span class="n">dds</span><span class="p">,</span><span class="w"> </span><span class="n">normalized</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">3</span>
<span class="n">dds.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dds</span><span class="p">[</span><span class="n">idx</span><span class="p">,</span><span class="w"> </span><span class="p">]</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 12  6
</code></pre></div>
12 genes remaining</p>
<p>We perform the estimation of dispersions 
<div class="highlight"><pre><span></span><code><span class="n">dds.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DESeq</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
</code></pre></div></p>
<p>PCA plot of the samples:
<div class="highlight"><pre><span></span><code><span class="n">vsd</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">varianceStabilizingTransformation</span><span class="p">(</span><span class="n">dds.f</span><span class="p">,</span><span class="w"> </span><span class="n">blind</span><span class="o">=</span><span class="kc">TRUE</span><span class="w"> </span><span class="p">)</span>
<span class="n">pcaData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">plotPCA</span><span class="p">(</span><span class="n">vsd</span><span class="p">,</span><span class="w"> </span><span class="n">intgroup</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;treatment&quot;</span><span class="p">))</span>
<span class="n">pcaData</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_label</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">PC1</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">PC2</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">))</span>
</code></pre></div>
<img alt="pca no outliers" src="../../assets/images/DESeq2_mouseMT/mouseMT_pca2.png" /></p>
<p>It looks much better. Seems like PC1 captures the group effect</p>
<p>We plot the estimate of the dispersions
<div class="highlight"><pre><span></span><code><span class="c1"># * black dot : raw</span>
<span class="c1"># * red dot : local trend</span>
<span class="c1"># * blue : corrected</span>
<span class="nf">plotDispEsts</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>
</code></pre></div></p>
<p><img alt="dispersion estimate mouseMT" src="../../assets/images/DESeq2_mouseMT/mouseMT_dispEst.png" /></p>
<p>There is so few genes that this does not look super nice here</p>
<p>For the Ruhland2016 dataset it looks like:</p>
<p><img alt="dispersion estimate Ruhland2016" src="../../assets/images/DESeq2/ruhland2016_dispEst.png" /></p>
<p>This plot is not easy to interpret. It represents the amount of dispersion at different levels of expression. It is directly linked to our ability to detect differential expression.</p>
<p>Here it looks about normal compared to typical bulk RNA-seq experiments : the dispersion is comparatively larger for lowly-expressed genes.</p>
<p><div class="highlight"><pre><span></span><code><span class="c1"># extracting results for the treatment versus control contrast</span>
<span class="n">res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">results</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>out of 12 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1, 8.3%
LFC &lt; 0 (down)     : 2, 17%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results
</code></pre></div></p>
<p>We can have a look at the coefficients of this model
<div class="highlight"><pre><span></span><code><span class="nf">head</span><span class="p">(</span><span class="nf">coef</span><span class="p">(</span><span class="n">dds.f</span><span class="p">))</span><span class="w"> </span><span class="c1"># the second column corresponds to the difference between the 2 conditions</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>                   Intercept treatment_b_vs_a
ENSMUSG00000064341 12.084282      -3.33601412
ENSMUSG00000064345  6.112479      -0.99101369
ENSMUSG00000064351  3.757967       0.64546475
ENSMUSG00000064354 10.209339       1.44160442
ENSMUSG00000064357 11.763707      -0.06245774
ENSMUSG00000064358  6.026334      -0.26447858
</code></pre></div></p>
<p>Here, it contains an intercept and a coefficient for the difference between the two groups.</p>
<p>MA plot:
<div class="highlight"><pre><span></span><code><span class="n">res.lfc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lfcShrink</span><span class="p">(</span><span class="n">dds.f</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="n">res</span><span class="p">)</span>
<span class="n">DESeq2</span><span class="o">::</span><span class="nf">plotMA</span><span class="p">(</span><span class="n">res.lfc</span><span class="p">)</span>
</code></pre></div>
<img alt="MA plot mouseMT" src="../../assets/images/DESeq2_mouseMT/mouseMT_MA.png" /></p>
<p>Volcano plot:
<div class="highlight"><pre><span></span><code><span class="n">FDRthreshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.01</span>
<span class="n">logFCthreshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.0</span>
<span class="c1"># add a column of NAs</span>
<span class="n">res.lfc</span><span class="o">$</span><span class="n">diffexpressed</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;NO&quot;</span>
<span class="c1"># if log2Foldchange &gt; 1 and pvalue &lt; 0.01, set as &quot;UP&quot; </span>
<span class="n">res.lfc</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">[</span><span class="n">res.lfc</span><span class="o">$</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">logFCthreshold</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">res.lfc</span><span class="o">$</span><span class="n">padj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="n">FDRthreshold</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;UP&quot;</span>
<span class="c1"># if log2Foldchange &lt; 1 and pvalue &lt; 0.01, set as &quot;DOWN&quot;</span>
<span class="n">res.lfc</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">[</span><span class="n">res.lfc</span><span class="o">$</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="o">-</span><span class="n">logFCthreshold</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">res.lfc</span><span class="o">$</span><span class="n">padj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="n">FDRthreshold</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;DOWN&quot;</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="w"> </span><span class="n">res.lfc</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">padj</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="n">diffexpressed</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="o">-</span><span class="n">logFCthreshold</span><span class="p">,</span><span class="w"> </span><span class="n">logFCthreshold</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">FDRthreshold</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">))</span>

<span class="nf">table</span><span class="p">(</span><span class="n">res.lfc</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>DOWN   NO   UP 
   1   10    1 
</code></pre></div>
<img alt="volcano plot mouseMT" src="../../assets/images/DESeq2_mouseMT/mouseMT_volcano.png" /></p>
<p>Heatmap:
<div class="highlight"><pre><span></span><code><span class="n">vsd.counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">assay</span><span class="p">(</span><span class="n">vsd</span><span class="p">)</span>

<span class="n">topVarGenes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">head</span><span class="p">(</span><span class="nf">order</span><span class="p">(</span><span class="nf">rowVars</span><span class="p">(</span><span class="n">vsd.counts</span><span class="p">),</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="m">20</span><span class="p">)</span>
<span class="n">mat</span><span class="w">  </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">vsd.counts</span><span class="p">[</span><span class="w"> </span><span class="n">topVarGenes</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="w"> </span><span class="c1">#scaled counts of the top genes</span>
<span class="n">mat</span><span class="w">  </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mat</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span><span class="w">  </span><span class="c1"># centering</span>
<span class="nf">pheatmap</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>
</code></pre></div>
<img alt="heatmap plot mouseMT" src="../../assets/images/DESeq2_mouseMT/mouseMT_heatmap.png" /></p>
<h2 id="saving-results-to-file">saving results to file</h2>
<p>note: a CSV file can be imported into Excel
<div class="highlight"><pre><span></span><code><span class="nf">write.csv</span><span class="p">(</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="p">,</span><span class="s">&#39;mouseMT.DESeq2.results.csv&#39;</span><span class="w"> </span><span class="p">)</span>
</code></pre></div></p>
</details>
<details class="note">
<summary>edgeR analysis</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># setup</span>
<span class="nf">library</span><span class="p">(</span><span class="n">edgeR</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
</code></pre></div>
<h2 id="reading_1">reading</h2>
<p>We will use the trimmed reads</p>
<p>First we try with the sample_a1:</p>
<p><div class="highlight"><pre><span></span><code><span class="n">folder</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/shared/data/Solutions/mouseMT/044_STAR_map_trimmed/&quot;</span>

<span class="c1"># we skip the 4 first lines, which contains </span>
<span class="c1"># N_unmapped , N_multimapping , N_noFeature , N_ambiguous   </span>

<span class="n">sample_a1_table</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="w"> </span><span class="n">folder</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sample_a1&quot;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.ReadsPerGene.out.tab&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">skip</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="w"> </span><span class="n">sample_a1_table</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>                    V2      V3      V4
ENSMUSG00000064336  0       0       0   
ENSMUSG00000064337  0       0       0   
ENSMUSG00000064338  0       0       0   
ENSMUSG00000064339  0       0       0   
ENSMUSG00000064340  0       0       0   
ENSMUSG00000064341  4046    1991    2055    
</code></pre></div></p>
<p>We are interested in the first columns, which contains counts for unstranded reads</p>
<p>Let&rsquo;s use a loop to automatize the reading:</p>
<p><div class="highlight"><pre><span></span><code><span class="n">raw_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="nf">row.names</span><span class="p">(</span><span class="n">sample_a1_table</span><span class="p">)</span><span class="w"> </span><span class="p">)</span>

<span class="nf">for</span><span class="p">(</span><span class="w"> </span><span class="n">sample</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;a1&#39;</span><span class="p">,</span><span class="s">&#39;a2&#39;</span><span class="p">,</span><span class="s">&#39;a3&#39;</span><span class="p">,</span><span class="s">&#39;a4&#39;</span><span class="p">,</span><span class="s">&#39;b1&#39;</span><span class="p">,</span><span class="s">&#39;b2&#39;</span><span class="p">,</span><span class="s">&#39;b3&#39;</span><span class="p">,</span><span class="s">&#39;b4&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">){</span>
<span class="w">  </span><span class="n">sample_table</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="w"> </span><span class="n">folder</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sample_&quot;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">sample</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.ReadsPerGene.out.tab&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span>
<span class="w">                            </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">skip</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="p">)</span>

<span class="w">  </span><span class="n">raw_counts</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sample_table</span><span class="p">[</span><span class="w"> </span><span class="nf">row.names</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&quot;V2&quot;</span><span class="w"> </span><span class="p">]</span>

<span class="p">}</span>

<span class="nf">head</span><span class="p">(</span><span class="w"> </span><span class="n">raw_counts</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>                    a1      a2      a3      a4  b1  b2  b3  b4
ENSMUSG00000064336  0       0       0       0   0   0   0   0
ENSMUSG00000064337  0       0       0       0   0   0   0   0
ENSMUSG00000064338  0       0       0       0   0   0   0   0
ENSMUSG00000064339  0       0       0       2   0   0   0   0
ENSMUSG00000064340  0       0       0       0   0   0   0   0
ENSMUSG00000064341  4046    4098    4031    1   449 515 13  456
</code></pre></div></p>
<h2 id="experimental-design">experimental design</h2>
<blockquote>
<p>note: levels lets us define the reference levels</p>
</blockquote>
<p><div class="highlight"><pre><span></span><code><span class="n">treatment</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;a&quot;</span><span class="p">,</span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;b&quot;</span><span class="p">,</span><span class="m">4</span><span class="p">)),</span><span class="w"> </span><span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;a&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;b&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">treatment</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span>

<span class="n">treatment</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>a1 a2 a3 a4 b1 b2 b3 b4 
 a  a  a  a  b  b  b  b 
Levels: a b
</code></pre></div></p>
<h2 id="edger-object-preprocessing-and-qc">edgeR  object preprocessing and QC</h2>
<p>Creating the edgeR DGE object and filtering low-count genes.</p>
<p><div class="highlight"><pre><span></span><code><span class="n">dge.all</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DGEList</span><span class="p">(</span><span class="n">counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span><span class="w">  </span>

<span class="n">dge.f.design</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">model.matrix</span><span class="p">(</span><span class="o">~</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span>

<span class="c1"># filtering by expression level. See ?filterByExpr for details</span>
<span class="n">keep</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filterByExpr</span><span class="p">(</span><span class="n">dge.all</span><span class="p">)</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dge.all</span><span class="p">[</span><span class="n">keep</span><span class="p">,</span><span class="w"> </span><span class="n">keep.lib.sizes</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">]</span>
<span class="nf">table</span><span class="p">(</span><span class="w"> </span><span class="n">keep</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>keep
FALSE  TRUE 
   28     9 
</code></pre></div></p>
<p><div class="highlight"><pre><span></span><code><span class="c1">#normalization</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">calcNormFactors</span><span class="p">(</span><span class="n">dge.f</span><span class="p">)</span>
<span class="n">dge.f</span><span class="o">$</span><span class="n">samples</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>    group   lib.size    norm.factors
a1  a       13799       1.2101311   
a2  a       13649       1.2130900   
a3  a       13938       1.2131513   
a4  a       6831        0.1058474   
b1  b       13703       1.3614563   
b2  b       13627       1.3728761   
b3  b       162         2.0759281   
b4  b       13687       1.3671996   
</code></pre></div>
We represent the distances between the samples using MDS:</p>
<p><div class="highlight"><pre><span></span><code><span class="nf">plotMDS</span><span class="p">(</span><span class="w"> </span><span class="n">dge.f</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;cornflowerblue&#39;</span><span class="p">,</span><span class="s">&#39;forestgreen&#39;</span><span class="p">)[</span><span class="n">treatment</span><span class="p">]</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<img alt="mouseMT MDS plot all samples" src="../../assets/images/edgeR_mouseMT/mouseMT_MDS1.png" /></p>
<p>OK, so a4 and b3 are quite different from the rest.</p>
<ul>
<li>a4 was expected from the QC</li>
<li>b3 we did not expect until now</li>
</ul>
<p>If we did the analysis with them, here is what we get:
<div class="highlight"><pre><span></span><code><span class="c1"># estimate of the dispersion</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">estimateDisp</span><span class="p">(</span><span class="n">dge.f</span><span class="p">,</span><span class="n">dge.f.design</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">robust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="c1"># testing for differential expression. </span>
<span class="n">dge.f.et</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exactTest</span><span class="p">(</span><span class="n">dge.f</span><span class="p">)</span>
<span class="nf">topTags</span><span class="p">(</span><span class="n">dge.f.et</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>                    logFC       logCPM      PValue      FDR
ENSMUSG00000065947  -5.06512022 13.51727    0.006432974 0.05110489
ENSMUSG00000064351  -7.46603064 20.14951    0.011356643 0.05110489
ENSMUSG00000064345  3.35236454  15.28962    0.050902734 0.13212506
ENSMUSG00000064341  -2.67947447 16.69126    0.058722251 0.13212506
ENSMUSG00000064354  1.58127747  16.60293    0.263189690 0.42743645
ENSMUSG00000064368  1.75395362  12.51557    0.284957631 0.42743645
ENSMUSG00000064358  -0.10554311 11.49478    0.912516094 0.96610797
ENSMUSG00000064363  -0.08216781 17.90743    0.950808801 0.96610797
ENSMUSG00000064357  0.06455233  17.18842    0.966107973 0.96610797
</code></pre></div></p>
<p>no gene is significantly DE.</p>
<p>So, let&rsquo;s eliminate these two samples.</p>
<h2 id="analysis-without-the-outliers_1">analysis without the outliers</h2>
<p><div class="highlight"><pre><span></span><code><span class="n">raw_counts_no_outliers</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="p">[</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="o">!</span><span class="p">(</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;a4&#39;</span><span class="p">,</span><span class="s">&#39;b3&#39;</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">]</span>
<span class="nf">head</span><span class="p">(</span><span class="w"> </span><span class="n">raw_counts_no_outliers</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>                    a1      a2      a3      b1  b2  b4
ENSMUSG00000064336  0       0       0       0   0   0
ENSMUSG00000064337  0       0       0       0   0   0
ENSMUSG00000064338  0       0       0       0   0   0
ENSMUSG00000064339  0       0       0       0   0   0
ENSMUSG00000064340  0       0       0       0   0   0
ENSMUSG00000064341  4046    4098    4031    449 515 456
</code></pre></div></p>
<p><div class="highlight"><pre><span></span><code><span class="n">treatment</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;a&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;b&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">)),</span><span class="w"> </span><span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;a&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;b&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">)</span>
<span class="n">colData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">treatment</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">raw_counts_no_outliers</span><span class="p">))</span>
<span class="n">colData</span><span class="w"> </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>    treatment
a1  a           
a2  a           
a3  a           
b1  b           
b2  b           
b4  b
</code></pre></div></p>
<p><div class="highlight"><pre><span></span><code><span class="n">dge.all</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DGEList</span><span class="p">(</span><span class="n">counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts_no_outliers</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span><span class="w">  </span>

<span class="n">dge.f.design</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">model.matrix</span><span class="p">(</span><span class="o">~</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span>

<span class="c1"># filtering by expression level. See ?filterByExpr for details</span>
<span class="n">keep</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filterByExpr</span><span class="p">(</span><span class="n">dge.all</span><span class="p">)</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dge.all</span><span class="p">[</span><span class="n">keep</span><span class="p">,</span><span class="w"> </span><span class="n">keep.lib.sizes</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">]</span>
<span class="nf">table</span><span class="p">(</span><span class="w"> </span><span class="n">keep</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>keep
FALSE  TRUE 
   28     9 
</code></pre></div></p>
<p>We compute the normalization factor for each library:
<div class="highlight"><pre><span></span><code><span class="c1">#normalization</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">calcNormFactors</span><span class="p">(</span><span class="n">dge.f</span><span class="p">)</span>
<span class="n">dge.f</span><span class="o">$</span><span class="n">samples</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>    group   lib.size    norm.factors
a1  a       13799       0.9437444   
a2  a       13649       0.9277668   
a3  a       13938       0.9412032   
b1  b       13703       1.0672149   
b2  b       13627       1.0651230   
b4  b       13687       1.0675095   
</code></pre></div></p>
<p>We represent the distances between the samples using MDS:</p>
<p><div class="highlight"><pre><span></span><code><span class="nf">plotMDS</span><span class="p">(</span><span class="w"> </span><span class="n">dge.f</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;cornflowerblue&#39;</span><span class="p">,</span><span class="s">&#39;forestgreen&#39;</span><span class="p">)[</span><span class="n">treatment</span><span class="p">]</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<img alt="mouseMT MDS plot no outliers" src="../../assets/images/edgeR_mouseMT/mouseMT_MDS2.png" /></p>
<p>It looks much better. Seems like PC1 captures the group effect.</p>
<p>We now fit the model:</p>
<p><div class="highlight"><pre><span></span><code><span class="c1"># estimate of the dispersion</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">estimateDisp</span><span class="p">(</span><span class="n">dge.f</span><span class="p">,</span><span class="n">dge.f.design</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">robust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="nf">plotBCV</span><span class="p">(</span><span class="n">dge.f</span><span class="p">)</span>
</code></pre></div>
<img alt="mouseMT BCV" src="../../assets/images/edgeR_mouseMT/mouseMT_BCV.png" /></p>
<p>There are so few genes that this does not look super nice here.</p>
<p>Here is how it looks like on the Ruhland2016 data:</p>
<p><img alt="bcv" src="../../assets/images/edgeR/BCV.png" /></p>
<p>This plot is not easy to interpret. It represents the amount of biological variation at different levels of expression. It is directly linked to our ability to detect differential expression.</p>
<p>Here it looks about normal compared to other bulk RNA-seq experiments : the variation is comparatively larger for lowly expressed genes.</p>
<p><div class="highlight"><pre><span></span><code><span class="c1"># testing for differential expression. </span>
<span class="c1"># This method is recommended when you only have 2 groups to compare</span>
<span class="n">dge.f.et</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exactTest</span><span class="p">(</span><span class="n">dge.f</span><span class="p">)</span>
<span class="nf">topTags</span><span class="p">(</span><span class="n">dge.f.et</span><span class="p">)</span><span class="w"> </span><span class="c1"># printing the genes where the p-value of differential expression if the lowest</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>                    logFC           logCPM      PValue          FDR
ENSMUSG00000064341  -3.2728209590   17.34360    0.000000e+00    0.000000e+00
ENSMUSG00000064354  1.5046106466    17.35950    0.000000e+00    0.000000e+00
ENSMUSG00000064345  -0.9251244495   11.92467    4.542291e-08    1.362687e-07
ENSMUSG00000064368  0.7262191019    10.55226    1.337410e-02    3.009174e-02
ENSMUSG00000064351  0.7031516899    10.50496    2.006905e-02    3.612429e-02
ENSMUSG00000064358  -0.1964770294   12.14888    2.110506e-01    3.165759e-01
ENSMUSG00000065947  0.2623477474    10.00748    4.831813e-01    5.851717e-01
ENSMUSG00000064363  -0.0127783014   18.61543    5.201527e-01    5.851717e-01
ENSMUSG00000064357  0.0006819028    17.93963    9.833826e-01    9.833826e-01
</code></pre></div></p>
<p>We can see 3 genes with FDR &lt; 0.01 and 2 others with 0.01 &lt; FDR &lt; 0.05.</p>
<p><div class="highlight"><pre><span></span><code><span class="nf">summary</span><span class="p">(</span><span class="nf">decideTests</span><span class="p">(</span><span class="n">dge.f.et</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">p.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.01</span><span class="p">))</span><span class="w"> </span><span class="c1"># let&#39;s use 0.01 as a threshold</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>        b-a
Down     2
NotSig   6
Up       1
</code></pre></div></p>
<p>Let&rsquo;s plot these:
<div class="highlight"><pre><span></span><code><span class="c1">## plot all the logFCs versus average count size. Significantly DE genes are  colored</span>
<span class="nf">plotMD</span><span class="p">(</span><span class="n">dge.f.et</span><span class="p">)</span>
<span class="c1"># lines at a log2FC of 1/-1, corresponding to a shift in expression of x2 </span>
<span class="nf">abline</span><span class="p">(</span><span class="n">h</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-1</span><span class="p">,</span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;blue&quot;</span><span class="p">)</span>
<span class="nf">abline</span><span class="p">(</span><span class="n">h</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;grey&quot;</span><span class="p">)</span>
</code></pre></div>
<img alt="mouseMT MA" src="../../assets/images/edgeR_mouseMT/mouseMT_MA.png" /></p>
<p>Volcano plot</p>
<p><div class="highlight"><pre><span></span><code><span class="n">allGenes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">topTags</span><span class="p">(</span><span class="n">dge.f.et</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">dge.f.et</span><span class="o">$</span><span class="n">table</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="o">$</span><span class="n">table</span>

<span class="n">FDRthreshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.01</span>
<span class="n">logFCthreshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.0</span>
<span class="c1"># add a column of NAs</span>
<span class="n">allGenes</span><span class="o">$</span><span class="n">diffexpressed</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;NO&quot;</span>
<span class="c1"># if log2Foldchange &gt; 1 and pvalue &lt; 0.01, set as &quot;UP&quot; </span>
<span class="n">allGenes</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">[</span><span class="n">allGenes</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">logFCthreshold</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">allGenes</span><span class="o">$</span><span class="n">FDR</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="n">FDRthreshold</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;UP&quot;</span>
<span class="c1"># if log2Foldchange &lt; 1 and pvalue &lt; 0.01, set as &quot;DOWN&quot;</span>
<span class="n">allGenes</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">[</span><span class="n">allGenes</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="o">-</span><span class="n">logFCthreshold</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">allGenes</span><span class="o">$</span><span class="n">FDR</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="n">FDRthreshold</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;DOWN&quot;</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">allGenes</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="n">logFC</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">FDR</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="n">diffexpressed</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="o">-</span><span class="n">logFCthreshold</span><span class="p">,</span><span class="w"> </span><span class="n">logFCthreshold</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">FDRthreshold</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">))</span>
</code></pre></div>
<img alt="mouseMT volcano" src="../../assets/images/edgeR_mouseMT/mouseMT_volcano.png" /></p>
<h2 id="writing-the-table-of-results">writing the table of results</h2>
<div class="highlight"><pre><span></span><code><span class="nf">write.csv</span><span class="p">(</span><span class="w"> </span><span class="n">allGenes</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&#39;mouseMT.edgeR.results.csv&#39;</span><span class="p">)</span>
</code></pre></div>
</details>
<div class="admonition note">
<p class="admonition-title">From ensembl gene ids to gene names</p>
<p>We can convert between different gene ids using the <code>bitr</code> function from <code>clusterProfiler</code>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">org.Mm.eg.db</span><span class="p">)</span>

<span class="n">genes_universe</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bitr</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">allGenes</span><span class="p">),</span><span class="w"> </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
<span class="w">                       </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">),</span>
<span class="w">                       </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;org.Mm.eg.db&quot;</span><span class="p">)</span>
<span class="n">genes_universe</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>             ENSEMBL ENTREZID SYMBOL
1 ENSMUSG00000064341    17716    ND1
2 ENSMUSG00000064354    17709   COX2
3 ENSMUSG00000064345    17717    ND2
4 ENSMUSG00000064368    17722    ND6
5 ENSMUSG00000064351    17708   COX1
6 ENSMUSG00000064358    17710   COX3
7 ENSMUSG00000065947    17720   ND4L
8 ENSMUSG00000064363    17719    ND4
9 ENSMUSG00000064357    17705   ATP6
</code></pre></div></p>
<blockquote>
<p>Here is the list of <a href="https://bioconductor.org/packages/release/BiocViews.html#___OrgDb">orgDb packages</a>. For non-model organisms it will be more complex.</p>
</blockquote>
</div>
<h2 id="differential-expression-task">Differential Expression - Task</h2>
<p>Use either edgeR or DESeq2 to conduct a differential expression analysis.</p>
<p>You may play with either of the following datasets:</p>
<ul>
<li>Ruhland2016<ul>
<li>simple 1 factor design</li>
<li><code>/shared/data/Solutions/Ruhland2016/countFiles/featureCounts_Ruhland2016.counts.txt</code></li>
<li><a :=":" class="md-button" href="../../assets/txt/featureCounts_Ruhland2016.counts.txt" target="_blank"> <span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0h160v128c0 17.7 14.3 32 32 32h128v288c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0l128 128z"/></svg></span> Ruhland2016 count matrix</a></li>
</ul>
</li>
<li>the Liu2015 dataset:<ul>
<li>simple 1 factor design</li>
<li><code>/shared/data/Solutions/Liu2015/countFiles/featureCounts_Liu2015.counts.txt</code></li>
<li><a :=":" class="md-button" href="../../assets/txt/featureCounts_Liu2015.counts.txt" target="_blank"> <span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0h160v128c0 17.7 14.3 32 32 32h128v288c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0l128 128z"/></svg></span> Liu2015 count matrix</a></li>
</ul>
</li>
<li><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0009317">Tuch 2010</a> dataset<ul>
<li>2 factors design : 3 patients (8, 33, and 51) each had 1 sample from tumor tissue (T) and normal tissue (N) sequenced. </li>
<li>the goal is to find the difference between tumor and normal while taking the patient into account.</li>
<li><code>/shared/data/Solutions/Tuch2010/Tuch_et_al_2010_counts.csv</code></li>
<li><a :=":" class="md-button" href="../../assets/txt/Tuch_et_al_2010_counts.csv" target="_blank"> <span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0h160v128c0 17.7 14.3 32 32 32h128v288c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0l128 128z"/></svg></span> Tuch 2010 count matrix</a></li>
</ul>
</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul>
<li>Generally, users find the syntax and workflow of DESeq2 easier for getting started.</li>
<li>If you have the time, conduct a differential expression analysis using both DESeq2 and edgeR.</li>
<li>
<p>Follow the vignettes/user&rsquo;s guide! They are the most up-to-date documents, and generally contain everything a newcomer might need, including worked-out examples.</p>
</li>
<li>
<p>when dealing with more than one factor, you will need a <strong>model matrix</strong> to specify the experimental design to the library, and to craft your <strong>contrasts</strong> of interest. The <a href="https://csoneson.github.io/ExploreModelMatrix/index.html">ExploreModelMatrix</a> package may help you a lot in that regard.</p>
</li>
</ul>
</div>
<h3 id="ruhland2016-deseq2-correction">Ruhland2016 - DESeq2 correction</h3>
<p><a :=":" class="md-button" href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html" target="_blank">DESeq2 vignette</a></p>
<details class="success">
<summary>read in the data</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># setup</span>
<span class="nf">library</span><span class="p">(</span><span class="n">DESeq2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>


<span class="c1"># reading the counts files - adapt the file path to your situation</span>
<span class="n">raw_counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;/shared/data/Solutions/Ruhland2016/countFiles/featureCounts_Ruhland2016.counts.txt&#39;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span>
<span class="w">                        </span><span class="n">skip</span><span class="o">=</span><span class="m">1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>

<span class="c1"># setting up row names as ensembl gene ids</span>
<span class="nf">row.names</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="o">$</span><span class="n">Geneid</span>

<span class="c1">## looking at the beginning of that table</span>
<span class="n">raw_counts</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>

<span class="c1"># removing these first columns to keep only the sample counts</span>
<span class="n">raw_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="p">[</span><span class="w"> </span><span class="p">,</span><span class="w">  </span><span class="m">-1</span><span class="o">:</span><span class="m">-6</span><span class="w">  </span><span class="p">]</span><span class="w"> </span>


<span class="c1"># changing column names</span>
<span class="nf">names</span><span class="p">(</span><span class="w"> </span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;_.*&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;.*.SRR[0-9]{7}_&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">)</span>

<span class="c1"># some checking of what we just read</span>
<span class="nf">head</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">);</span><span class="w"> </span><span class="nf">tail</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">);</span><span class="w"> </span><span class="nf">dim</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span>
<span class="nf">colSums</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="c1"># total number of counted reads per sample</span>
</code></pre></div>
</details>
<details class="success">
<summary>preprocessing</summary>
<div class="highlight"><pre><span></span><code><span class="c1">## telling DESeq2 what the experimental design was</span>
<span class="c1"># note: by default, the 1st level is considered to be the reference/control/WT/...</span>
<span class="n">treatment</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;EtOH&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;TAM&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">)),</span><span class="w"> </span><span class="n">levels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;EtOH&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TAM&quot;</span><span class="p">)</span><span class="w"> </span><span class="p">)</span>
<span class="n">colData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">treatment</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">))</span>
<span class="n">colData</span>

<span class="c1">## creating the DESeq data object &amp; positing the model</span>
<span class="n">dds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DESeqDataSetFromMatrix</span><span class="p">(</span>
<span class="w">  </span><span class="n">countData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="p">,</span><span class="w"> </span><span class="n">colData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colData</span><span class="p">,</span><span class="w"> </span>
<span class="w">  </span><span class="n">design</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>

<span class="c1">## filter low count genes. Here, only keep genes with at least 2 samples where there are at least 5 reads.</span>
<span class="n">idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowSums</span><span class="p">(</span><span class="nf">counts</span><span class="p">(</span><span class="n">dds</span><span class="p">,</span><span class="w"> </span><span class="n">normalized</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">2</span>
<span class="n">dds.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dds</span><span class="p">[</span><span class="n">idx</span><span class="p">,</span><span class="w"> </span><span class="p">]</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>

<span class="c1"># we go from 55414 to 19378 genes</span>
</code></pre></div>
<p>Around 19k genes pass our minimum expression threshold, quite typical for a bulk Mouse RNA-seq experiment.</p>
</details>
<details class="success">
<summary>estimate dispersion / model fitting</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># we perform the estimation of dispersions </span>
<span class="n">dds.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DESeq</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>

<span class="c1"># we plot the estimate of the dispersions</span>
<span class="c1"># * black dot : raw</span>
<span class="c1"># * red dot : local trend</span>
<span class="c1"># * blue : corrected</span>
<span class="nf">plotDispEsts</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>

<span class="c1"># extracting results for the treatment versus control contrast</span>
<span class="n">res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">results</span><span class="p">(</span><span class="n">dds.f</span><span class="p">)</span>
</code></pre></div>
<p><img alt="dispEst" src="../../assets/images/DESeq2/ruhland2016_dispEst.png" /></p>
<p>This plot is not easy to interpret. It represents the amount of dispersion at different levels of expression. It is directly linked to our ability to detect differential expression.</p>
<p>Here it looks about normal compared to typical bulk RNA-seq experiments : the dispersion is comparatively larger for lowly expressed genes.</p>
</details>
<details class="success">
<summary>looking at the results</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># adds estimate of the LFC the results table. </span>
<span class="c1"># This shrunk logFC estimate is more robust than the raw value</span>

<span class="nf">head</span><span class="p">(</span><span class="nf">coef</span><span class="p">(</span><span class="n">dds.f</span><span class="p">))</span><span class="w"> </span><span class="c1"># the second column corresponds to the difference between the 2 conditions</span>
<span class="n">res.lfc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lfcShrink</span><span class="p">(</span><span class="n">dds.f</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="n">res</span><span class="p">)</span>

<span class="c1">#plotting to see the difference.  </span>
<span class="nf">par</span><span class="p">(</span><span class="n">mfrow</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">1</span><span class="p">))</span>
<span class="n">DESeq2</span><span class="o">::</span><span class="nf">plotMA</span><span class="p">(</span><span class="n">res</span><span class="p">)</span>
<span class="n">DESeq2</span><span class="o">::</span><span class="nf">plotMA</span><span class="p">(</span><span class="n">res.lfc</span><span class="p">)</span>
<span class="c1"># -&gt; with shrinkage, the significativeness and logFC are more consistent</span>
<span class="nf">par</span><span class="p">(</span><span class="n">mfrow</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">))</span>
</code></pre></div>
<p><img alt="doubleMA" src="../../assets/images/DESeq2/ruhland2016_doubleMA.png" /></p>
<p>Without the shrinkage, we can see that for low counts we can see a high log-fold change but non significant (ie. we see a large difference but with variance is also so high that this observation may be due to chance only).</p>
<p>The shrinkage corrects this and the relationship between logFC and significance is smoother.</p>
<p><div class="highlight"><pre><span></span><code><span class="c1"># we apply the variance stabilising transformation to make the read counts comparable across libraries</span>
<span class="c1"># (nb : this is not needed for DESeq DE analysis, but rather for visualisations that compare expression across samples, such as PCA. This replaces normal PCA scaling)</span>
<span class="n">vst.dds.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vst</span><span class="p">(</span><span class="n">dds.f</span><span class="p">,</span><span class="w"> </span><span class="n">blind</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="n">vst.dds.f.counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">assay</span><span class="p">(</span><span class="n">vst.dds.f</span><span class="p">)</span>

<span class="nf">plotPCA</span><span class="p">(</span><span class="n">vst.dds.f</span><span class="p">,</span><span class="w"> </span><span class="n">intgroup</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;treatment&quot;</span><span class="p">))</span>
</code></pre></div>
<img alt="pca" src="../../assets/images/DESeq2/ruhland2016_PCA.png" /></p>
<p>The first axis (58% of the variance) seems linked to the grouping of interest.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## ggplot2-based volcano plot</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>

<span class="n">FDRthreshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.01</span>
<span class="n">logFCthreshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.0</span>
<span class="c1"># add a column of NAs</span>
<span class="n">res.lfc</span><span class="o">$</span><span class="n">diffexpressed</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;NO&quot;</span>
<span class="c1"># if log2Foldchange &gt; 1 and pvalue &lt; 0.01, set as &quot;UP&quot; </span>
<span class="n">res.lfc</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">[</span><span class="n">res.lfc</span><span class="o">$</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">logFCthreshold</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">res.lfc</span><span class="o">$</span><span class="n">padj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="n">FDRthreshold</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;UP&quot;</span>
<span class="c1"># if log2Foldchange &lt; 1 and pvalue &lt; 0.01, set as &quot;DOWN&quot;</span>
<span class="n">res.lfc</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">[</span><span class="n">res.lfc</span><span class="o">$</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="o">-</span><span class="n">logFCthreshold</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">res.lfc</span><span class="o">$</span><span class="n">padj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="n">FDRthreshold</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;DOWN&quot;</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="w"> </span><span class="n">res.lfc</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">padj</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="n">diffexpressed</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="o">-</span><span class="n">logFCthreshold</span><span class="p">,</span><span class="w"> </span><span class="n">logFCthreshold</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">FDRthreshold</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">))</span>

<span class="nf">table</span><span class="p">(</span><span class="n">res.lfc</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code> DOWN    NO    UP 
  131 19002   245 
</code></pre></div>
<p><img alt="volcano" src="../../assets/images/DESeq2/ruhland2016_volcano.png" /></p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">pheatmap</span><span class="p">)</span>
<span class="n">topVarGenes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">head</span><span class="p">(</span><span class="nf">order</span><span class="p">(</span><span class="nf">rowVars</span><span class="p">(</span><span class="n">vst.dds.f.counts</span><span class="p">),</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="m">20</span><span class="p">)</span>
<span class="n">mat</span><span class="w">  </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">vst.dds.f.counts</span><span class="p">[</span><span class="w"> </span><span class="n">topVarGenes</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="w"> </span><span class="c1">#scaled counts of the top genes</span>
<span class="n">mat</span><span class="w">  </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mat</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span><span class="w">  </span><span class="c1"># centering</span>
<span class="nf">pheatmap</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>
</code></pre></div>
<p><img alt="pheatmap" src="../../assets/images/DESeq2/ruhland2016_pheatmap.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># saving results to file</span>
<span class="c1"># note: a CSV file can be imported into Excel</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="p">,</span><span class="s">&#39;Ruhland2016.DESeq2.results.csv&#39;</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
</details>
<h3 id="ruhland2016-edger-correction">Ruhland2016 - EdgeR correction</h3>
<p><a :=":" class="md-button" href="https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf" target="_blank">edgeR user&rsquo;s guide</a></p>
<details class="success">
<summary>read in the data</summary>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">edgeR</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>

<span class="c1"># reading the counts files - adapt the file path to your situation</span>
<span class="n">raw_counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;.../Ruhland2016_featureCount_result.counts&#39;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">skip</span><span class="o">=</span><span class="m">1</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>

<span class="c1"># setting up row names as ensembl gene ids</span>
<span class="nf">row.names</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="o">$</span><span class="n">Geneid</span>

<span class="c1"># removing these first columns to keep only the sample counts</span>
<span class="n">raw_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="p">[</span><span class="w"> </span><span class="p">,</span><span class="w">  </span><span class="m">-1</span><span class="o">:</span><span class="m">-6</span><span class="w">  </span><span class="p">]</span><span class="w"> </span>

<span class="c1"># changing column names</span>
<span class="nf">names</span><span class="p">(</span><span class="w"> </span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;_.*&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;.*.SRR[0-9]{7}_&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">)</span>

<span class="c1"># some checking of what we just read</span>
<span class="nf">head</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">);</span><span class="w"> </span><span class="nf">tail</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">);</span><span class="w"> </span><span class="nf">dim</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span>
<span class="nf">colSums</span><span class="p">(</span><span class="n">raw_counts</span><span class="p">)</span><span class="w"> </span><span class="c1"># total number of counted reads per sample</span>
</code></pre></div>
</details>
<details class="success">
<summary>edgeR object preprocessing</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># setting up the experimental design AND the model</span>
<span class="c1">#  -&gt; the first 3 samples form a group, the 3 remaining are the other group</span>
<span class="n">treatment</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">  </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;EtOH&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;TAM&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">))</span>
<span class="n">dge.f.design</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">model.matrix</span><span class="p">(</span><span class="o">~</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span>

<span class="c1"># creating the edgeR DGE object</span>
<span class="n">dge.all</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DGEList</span><span class="p">(</span><span class="n">counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">raw_counts</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">treatment</span><span class="p">)</span><span class="w">  </span>

<span class="c1"># filtering by expression level. See ?filterByExpr for details</span>
<span class="n">keep</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filterByExpr</span><span class="p">(</span><span class="n">dge.all</span><span class="p">)</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dge.all</span><span class="p">[</span><span class="n">keep</span><span class="p">,</span><span class="w"> </span><span class="n">keep.lib.sizes</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">]</span>
<span class="nf">table</span><span class="p">(</span><span class="w"> </span><span class="n">keep</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>keep
FALSE  TRUE 
39702 15712 
</code></pre></div>
<p>Around 16k genes are sufficiently expressed to be retained.</p>
<div class="highlight"><pre><span></span><code><span class="c1">#normalization</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">calcNormFactors</span><span class="p">(</span><span class="n">dge.f</span><span class="p">)</span>
<span class="n">dge.f</span><span class="o">$</span><span class="n">samples</span>
</code></pre></div>
<p>Each sample has been associated with a normalization factor.</p>
</details>
<details class="success">
<summary>edgeR model fitting</summary>
<p><div class="highlight"><pre><span></span><code><span class="c1"># estimate of the dispersion</span>
<span class="n">dge.f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">estimateDisp</span><span class="p">(</span><span class="n">dge.f</span><span class="p">,</span><span class="n">dge.f.design</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">robust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="nf">plotBCV</span><span class="p">(</span><span class="n">dge.f</span><span class="p">)</span>
</code></pre></div>
<img alt="bcv" src="../../assets/images/edgeR/BCV.png" /></p>
<p>This plot is not easy to interpret. It represents the amount of biological variation at different levels of expression. It is directly linked to our ability to detect differential expression.</p>
<p>Here it looks about normal compared to other bulk RNA-seq experiments : the variation is comparatively larger for lowly expressed genes.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># testing for differential expression. </span>
<span class="c1"># This method is recommended when you only have 2 groups to compare</span>
<span class="n">dge.f.et</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exactTest</span><span class="p">(</span><span class="n">dge.f</span><span class="p">)</span>
<span class="nf">topTags</span><span class="p">(</span><span class="n">dge.f.et</span><span class="p">)</span><span class="w"> </span><span class="c1"># printing the genes where the p-value of differential expression if the lowest</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>Comparison of groups:  TAM-EtOH 
                       logFC   logCPM       PValue          FDR
ENSMUSG00000050272 -8.522762 4.988067 2.554513e-28 3.851950e-24
ENSMUSG00000075014  3.890079 5.175181 2.036909e-25 1.535728e-21
ENSMUSG00000009185  3.837786 6.742422 1.553964e-22 7.810743e-19
ENSMUSG00000075015  3.778523 3.274463 2.106799e-22 7.942107e-19
ENSMUSG00000028339 -5.692069 6.372980 4.593720e-16 1.385374e-12
ENSMUSG00000040111 -2.141221 6.771538 4.954522e-15 1.245154e-11
ENSMUSG00000041695  4.123972 1.668247 6.057909e-15 1.304960e-11
ENSMUSG00000072941  3.609170 7.080257 1.807618e-14 3.407135e-11
ENSMUSG00000000120 -6.340146 6.351489 2.507019e-14 4.200371e-11
ENSMUSG00000034981  3.727969 5.244841 3.934957e-14 5.933521e-11
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># see how many genes are DE</span>
<span class="nf">summary</span><span class="p">(</span><span class="nf">decideTests</span><span class="p">(</span><span class="n">dge.f.et</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">p.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.01</span><span class="p">))</span><span class="w"> </span><span class="c1"># let&#39;s use 0.01 as a threshold</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>         TAM-EtOH 
Down     109
NotSig 15393
Up       210
</code></pre></div>
<p>The comparison is TAM-EtOH, so &ldquo;Up&rdquo;, corresponds to a higher in group TAM compared to group EtOH.</p>
</details>
<details class="success">
<summary>edgeR looking at differentially-expressed genes</summary>
<div class="highlight"><pre><span></span><code><span class="c1">## plot all the logFCs versus average count size. Significantly DE genes are  colored</span>
<span class="nf">par</span><span class="p">(</span><span class="n">mfrow</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">))</span>
<span class="nf">plotMD</span><span class="p">(</span><span class="n">dge.f.et</span><span class="p">)</span>
<span class="c1"># lines at a log2FC of 1/-1, corresponding to a shift in expression of x2 </span>
<span class="nf">abline</span><span class="p">(</span><span class="n">h</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-1</span><span class="p">,</span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;blue&quot;</span><span class="p">)</span><span class="w"> </span>
</code></pre></div>
<p><img alt="edgeR_mdplot" src="../../assets/images/edgeR/MDplot.png" /></p>
<p><div class="highlight"><pre><span></span><code><span class="c1">## Volcano plot</span>
<span class="n">allGenes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">topTags</span><span class="p">(</span><span class="n">dge.f.et</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">dge.f.et</span><span class="o">$</span><span class="n">table</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="o">$</span><span class="n">table</span>

<span class="n">FDRthreshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.01</span>
<span class="n">logFCthreshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.0</span>
<span class="c1"># add a column of NAs</span>
<span class="n">allGenes</span><span class="o">$</span><span class="n">diffexpressed</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;NO&quot;</span>
<span class="c1"># if log2Foldchange &gt; 1 and pvalue &lt; 0.01, set as &quot;UP&quot; </span>
<span class="n">allGenes</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">[</span><span class="n">allGenes</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">logFCthreshold</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">allGenes</span><span class="o">$</span><span class="n">FDR</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="n">FDRthreshold</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;UP&quot;</span>
<span class="c1"># if log2Foldchange &lt; 1 and pvalue &lt; 0.01, set as &quot;DOWN&quot;</span>
<span class="n">allGenes</span><span class="o">$</span><span class="n">diffexpressed</span><span class="p">[</span><span class="n">allGenes</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="o">-</span><span class="n">logFCthreshold</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">allGenes</span><span class="o">$</span><span class="n">FDR</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="n">FDRthreshold</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;DOWN&quot;</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">allGenes</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="n">logFC</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">FDR</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="n">diffexpressed</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="o">-</span><span class="n">logFCthreshold</span><span class="p">,</span><span class="w"> </span><span class="n">logFCthreshold</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">FDRthreshold</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">))</span>
</code></pre></div>
<img alt="edgeR_volcano" src="../../assets/images/edgeR/ruhland2016_volcano.png" /></p>
<div class="highlight"><pre><span></span><code><span class="c1">## writing the table of results</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="w"> </span><span class="n">allGenes</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Ruhland2016.edgeR.results.csv&#39;</span><span class="p">)</span>
</code></pre></div>
</details>
<details class="success">
<summary>edgeR extra stuff</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># how to extract log CPM</span>
<span class="n">logcpm</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">cpm</span><span class="p">(</span><span class="n">dge.f</span><span class="p">,</span><span class="w"> </span><span class="n">prior.count</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">log</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># there is another fitting method reliying on quasi-likelihood, which is useful when the model is more complex (ie. more than 1 factor with 2 levels)</span>
<span class="n">dge.f.QLfit</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glmQLFit</span><span class="p">(</span><span class="n">dge.f</span><span class="p">,</span><span class="w"> </span><span class="n">dge.f.design</span><span class="p">)</span>
<span class="n">dge.f.qlt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glmQLFTest</span><span class="p">(</span><span class="n">dge.f.QLfit</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="o">=</span><span class="m">2</span><span class="p">)</span>

<span class="c1"># you can see the results are relatively different. The order of genes changes a bit, and the p-values are more profoundly affected</span>
<span class="nf">topTags</span><span class="p">(</span><span class="n">dge.f.et</span><span class="p">)</span>
<span class="nf">topTags</span><span class="p">(</span><span class="n">dge.f.qlt</span><span class="p">)</span>

<span class="c1">## let&#39;s see how much the two methods agree:</span>
<span class="nf">par</span><span class="p">(</span><span class="n">mfrow</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">))</span>
<span class="nf">plot</span><span class="p">(</span><span class="w"> </span><span class="n">dge.f.et</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="p">,</span><span class="w"> </span>
<span class="w">      </span><span class="n">dge.f.qlt</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">logFC</span><span class="p">,</span>
<span class="w">      </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;exact test logFC&#39;</span><span class="p">,</span>
<span class="w">      </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;quasi-likelihood test logFC&#39;</span><span class="p">)</span>

<span class="nf">print</span><span class="p">(</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&#39;logFC pearson correlation coefficient :&#39;</span><span class="w"> </span><span class="p">,</span><span class="w"> </span>
<span class="w">             </span><span class="nf">cor</span><span class="p">(</span><span class="n">dge.f.et</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="p">,</span><span class="n">dge.f.qlt</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">logFC</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">)</span>

<span class="nf">plot</span><span class="p">(</span><span class="w"> </span><span class="nf">log10</span><span class="p">(</span><span class="n">dge.f.et</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">PValue</span><span class="w"> </span><span class="p">),</span><span class="w"> </span>
<span class="w">      </span><span class="nf">log10</span><span class="p">(</span><span class="n">dge.f.qlt</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">PValue</span><span class="p">)</span><span class="w"> </span><span class="p">,</span>
<span class="w">      </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;exact test p-values (log10)&#39;</span><span class="p">,</span>
<span class="w">      </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;quasi-likelihood test p-values (log10)&#39;</span><span class="p">)</span>

<span class="nf">print</span><span class="p">(</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="w"> </span><span class="s">&quot;P-values spearman correlation coefficient&quot;</span><span class="p">,</span>
<span class="w">              </span><span class="nf">cor</span><span class="p">(</span><span class="w"> </span><span class="nf">log10</span><span class="p">(</span><span class="n">dge.f.et</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">PValue</span><span class="w"> </span><span class="p">),</span><span class="w"> </span><span class="nf">log10</span><span class="p">(</span><span class="n">dge.f.qlt</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">PValue</span><span class="p">)</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;spearman&#39;</span><span class="w"> </span><span class="p">)))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>&quot;logFC pearson correlation coefficient : 0.999997655536736&quot;
&quot;P-values spearman correlation coefficient 0.993238670517236&quot;
</code></pre></div>
<p><img alt="edgeR_compareTests" src="../../assets/images/edgeR/exact_QL_comparison.png" /></p>
<p>The logFC are highly correlated.
FDRs show less correlation but their <strong>ranks</strong> are highly correlated : they come in a very similar order.</p>
</details>
<h3 id="tuch-2010-edger-correction">Tuch 2010 - EdgeR correction</h3>
<p>We refer you here to section 4.1 of <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">edgeR&rsquo;s vignette</a>.</p>
<h2 id="additional-importing-counts-from-salmon-with-tximport">Additional : importing counts from salmon with <code>tximport</code></h2>
<p>The <code>tximport</code> R packages offers a fairly simple set of functions to get <strong>transcript-level</strong> expression quantification from salmon or kallisto into a differential <strong>gene</strong> expression analysis.</p>
<p><strong>Task :</strong> import salmon transcript-level quantification in R in order to perform a DE analysis on it using either edgeR or DESeq2.
<strong>Additional:</strong> compare the results with the ones obtained from STAR-aligned reads.</p>
<ul>
<li>The <a href="https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html">tximport vignette</a> is a very good guide for this task.</li>
<li>If you have not computed them, you can find files with expression quantifications in : <code>/shared/data/Solutions/Liu2015/</code> and <code>/shared/data/Solutions/Ruhland2016/</code></li>
</ul>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.74e28a9f.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.220ee61c.min.js"></script>
      
    
  </body>
</html>